<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chrysa Papadopoulou" />


<title>Definition of CCL19 FRC subsets and their interactions with immune cells in NSCLC</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CCL19_FRCs_lung_cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Definition of CCL19 FRC subsets and their
interactions with immune cells in NSCLC</h1>
<h4 class="author">Chrysa Papadopoulou</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-10-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CCL19_FRCs_lung_cancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240808code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240808)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240808code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240808)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomPchryssaCCL19FRCslungcancertreeb4bcf39d58b70e934768cc7ed3b235569088b121targetblankb4bcf39a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/tree/b4bcf39d58b70e934768cc7ed3b235569088b121" target="_blank">b4bcf39</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomPchryssaCCL19FRCslungcancertreeb4bcf39d58b70e934768cc7ed3b235569088b121targetblankb4bcf39a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/tree/b4bcf39d58b70e934768cc7ed3b235569088b121" target="_blank">b4bcf39</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    data/Final_submission/
    Ignored:    data/Human/
    Ignored:    data/Mouse/
    Ignored:    data/Public/
    Ignored:    output/GSEA_AdvFB_SULF1/
    Ignored:    output/GSEA_AdvFB_TLS/
    Ignored:    output/GSEA_CCR7_T/
    Ignored:    output/GSEA_CD8_T/
    Ignored:    output/GSEA_CYCL_T/
    Ignored:    output/GSEA_EXH_T/
    Ignored:    output/GSEA_SMC_PRC/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/.h5seurat
    Untracked:  analysis/Compare_tumors.Rmd
    Untracked:  analysis/NSCLC_PDAC_CAFs.Rmd
    Untracked:  analysis/QC_NSCLC.Rmd
    Untracked:  analysis/Seurat_to_SCE.Rmd
    Untracked:  analysis/compression.Rmd
    Untracked:  analysis/gene_metaprograms.Rmd
    Untracked:  analysis/index_hidden.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/comparison_CCL19_FRC_NSCLC_tonsil.Rmd</code>) and HTML
(<code>docs/comparison_CCL19_FRC_NSCLC_tonsil.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/b4bcf39d58b70e934768cc7ed3b235569088b121/analysis/comparison_CCL19_FRC_NSCLC_tonsil.Rmd" target="_blank">b4bcf39</a>
</td>
<td>
Pchryssa
</td>
<td>
2025-10-01
</td>
<td>
protocol define FRCs and signaling
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>suppressPackageStartupMessages({
  library(here)
  library(purrr)
  library(Seurat)
  library(dittoSeq)
  library(NMF)
  library(CellChat)
  library(harmony)
  library(ggsci)
  library(bigmds)
})</code></pre>
</div>
<div id="set-directory" class="section level2">
<h2>Set directory</h2>
<pre class="r"><code>basedir &lt;- here()</code></pre>
<div id="read-nsclc-ccl19-frc-data" class="section level3">
<h3>Read NSCLC CCL19 FRC data</h3>
<pre class="r"><code>NSCLC_CCL19_data &lt;- readRDS(paste0(basedir,&quot;/data/Human/NSCLC_CCL19_FRCs_CAFs.rds&quot;))</code></pre>
</div>
<div id="nsclc-ccl19-cells-in-nsclc" class="section level3">
<h3>NSCLC CCL19⁺ cells in NSCLC</h3>
<pre class="r"><code>#Define color palet
palet_CCL19_FRC &lt;- c(&quot;#1B9E77&quot;, &quot;#54B0E4&quot;,&quot;#E3BE00&quot;, &quot;#E41A1C&quot;)
names(palet_CCL19_FRC) &lt;- c(&quot;CAF2/TRC&quot;,&quot;CAF1/PRC&quot;,&quot;AdvFB&quot; ,&quot;SMC/PC&quot;)

palet_CCL19_FRC &lt;- palet_CCL19_FRC[names(palet_CCL19_FRC) %in% unique(NSCLC_CCL19_data$cell_type)]

DimPlot(NSCLC_CCL19_data, reduction = &quot;umap&quot;, group.by = &quot;cell_type&quot;,cols = palet_CCL19_FRC)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;) + ggtitle(paste0(&quot;CCL19&quot;, &quot;\U207A &quot;, &quot;fibroblasts&quot;))</code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/umap%20CCL19%20Cells-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="subset-on-ccl19-frcs" class="section level3">
<h3>Subset on CCL19⁺ FRCs</h3>
<pre class="r"><code>NCLS_FRCS &lt;- subset(NSCLC_CCL19_data, cell_type %in% c(&quot;CAF2/TRC&quot;,&quot;CAF1/PRC&quot;))

#Preprocessing
resolution &lt;- c(0.1, 0.25, 0.4, 0.6,0.7, 0.8, 0.9, 1.0, 1.2, 1.4, 1.6, 1.8, 2.0)
NCLS_FRCS &lt;- FindVariableFeatures(NCLS_FRCS, selection.method = &quot;vst&quot;, nfeatures = 2000)
NCLS_FRCS &lt;- ScaleData(NCLS_FRCS)
NCLS_FRCS &lt;- RunPCA(object = NCLS_FRCS, npcs = 30, verbose = FALSE,seed.use = 8734)
NCLS_FRCS &lt;- RunTSNE(object = NCLS_FRCS, reduction = &quot;pca&quot;, dims = 1:20, seed.use = 8734)
NCLS_FRCS &lt;- RunUMAP(object = NCLS_FRCS, reduction = &quot;pca&quot;, dims = 1:20, seed.use = 8734)
NCLS_FRCS &lt;- FindNeighbors(object = NCLS_FRCS, reduction = &quot;pca&quot;, dims = 1:20, seed.use = 8734)
for(k in 1:length(resolution)){
  NCLS_FRCS &lt;- FindClusters(object = NCLS_FRCS, resolution = resolution[k], random.seed = 8734)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9474
Number of communities: 4
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9089
Number of communities: 8
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8838
Number of communities: 9
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8560
Number of communities: 12
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8432
Number of communities: 13
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8315
Number of communities: 15
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8227
Number of communities: 16
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8139
Number of communities: 17
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7990
Number of communities: 17
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7839
Number of communities: 18
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7705
Number of communities: 21
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7591
Number of communities: 22
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 5023
Number of edges: 172182

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7483
Number of communities: 23
Elapsed time: 0 seconds</code></pre>
</div>
<div id="umap-of-re-embedded-ccl19-frc-subsets-figure-4b"
class="section level3">
<h3>Umap of Re-embedded CCL19⁺ FRC subsets (Figure 4B)</h3>
<pre class="r"><code>palet_CCL19_FRC &lt;- c(&quot;#5050FFFF&quot;, &quot;#CE3D32FF&quot;)
names(palet_CCL19_FRC) &lt;- c(&quot;CAF2/TRC&quot;,&quot;CAF1/PRC&quot;)

DimPlot(NCLS_FRCS, reduction = &quot;umap&quot;, group.by = &quot;cell_type&quot;,cols = palet_CCL19_FRC)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;) + ggtitle(paste0(&quot;CCL19&quot;, &quot;\U207A &quot;, &quot;Tumor FRCs&quot;))</code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/umap%20TRC%20PRC-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="tonsillar-frc-signatures-projected-on-tumor-ccl19-frc-subsets-figure-4c"
class="section level3 tabset">
<h3 class="tabset">Tonsillar FRC signatures projected on tumor CCL19⁺
FRC subsets (Figure 4C)</h3>
<div id="signature-slo-trc" class="section level4">
<h4>Signature SLO-TRC</h4>
<pre class="r"><code>NSCLC_TRC &lt;-c(&quot;CCL19&quot;,&quot;CCL21&quot;,&quot;PDPN&quot;,&quot;ICAM1&quot;,&quot;VCAM1&quot;,&quot;LUM&quot;,&quot;PDGFRA&quot;,&quot;TNFSF13B&quot;)
NSCLC_TRC &lt;- unlist(lapply(NSCLC_TRC, function(x) {
  get_full_gene_name(x,NCLS_FRCS)
}))
slot_type &lt;-&quot;data&quot;
gn &lt;- &quot;Tumor_TRC&quot;
Visualize_GeneSignatures_sc(NCLS_FRCS, NSCLC_TRC, slot_type, &#39;average.mean&#39;,gn) + ggtitle(&quot;SLO-TRC signature&quot;)</code></pre>
<pre><code>[1] &quot;gene.set.score_Tumor_TRC_data&quot;
Cell names order match in &#39;mean_expr&#39; and &#39;object@meta.data&#39;:
 adding gene set mean expression values in &#39;object@meta.data$gene.set.score&#39;</code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/signature%20SLO%20TRC-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="signature-slo-prc" class="section level4">
<h4>Signature SLO-PRC</h4>
<pre class="r"><code>NSCLC_PRC &lt;-c(&quot;CCL19&quot;,&quot;CCL21&quot;,&quot;ITGA1&quot;,&quot;ITGA7&quot;,&quot;MCAM&quot;,&quot;CNN1&quot;,&quot;NOTCH3&quot;,&quot;ACTA2&quot;,&quot;PDGFRB&quot;,&quot;ANGPT2&quot;)
NSCLC_PRC &lt;- unlist(lapply(NSCLC_PRC, function(x) {
  get_full_gene_name(x,NCLS_FRCS)
}))

slot_type &lt;-&quot;data&quot;
gn &lt;- &quot;Tumor_PRC&quot;
Visualize_GeneSignatures_sc(NCLS_FRCS, NSCLC_PRC, slot_type, &#39;average.mean&#39;,gn) + ggtitle(&quot;SLO-PRC signature&quot;)</code></pre>
<pre><code>[1] &quot;gene.set.score_Tumor_PRC_data&quot;
Cell names order match in &#39;mean_expr&#39; and &#39;object@meta.data&#39;:
 adding gene set mean expression values in &#39;object@meta.data$gene.set.score&#39;</code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/signature%20SLO%20PRC-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div
id="comparison-of-transcriptional-profiles-between-nsclc-ccl19-frcs-and-tonsilar-frcs-via-multidimensional-scaling-mds"
class="section level2">
<h2><strong>Comparison of transcriptional profiles between NSCLC CCL19⁺
FRCs and Tonsilar FRCs via Multidimensional scaling (MDS)</strong></h2>
<div id="read-tonsilar-frc-data" class="section level3">
<h3>Read Tonsilar FRC data</h3>
<pre class="r"><code>Tons_FRC_data &lt;-readRDS(paste0(basedir,&quot;/data/Human/mergedHumanTonsilExtendedDataset_incAcuteTonsilitis_mapped_wocl11+12+14_seuratFRC.rds&quot;))</code></pre>
</div>
<div id="define-color-palette" class="section level3">
<h3>Define color palette</h3>
<pre class="r"><code>cols&lt;- pal_igv()(51)
names(cols) &lt;- c(0:50)</code></pre>
</div>
<div id="tonsilar-frcs" class="section level3">
<h3>Tonsilar FRCs</h3>
<pre class="r"><code>Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;ACTA2+PRC_1&quot;] &lt;- paste0(&quot;ACTA2&quot;, expression(&quot;\U207A&quot;),&quot;PRC_1&quot;)
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;ACTA2+PRC_2&quot;] &lt;- paste0(&quot;ACTA2&quot;, expression(&quot;\U207A&quot;),&quot;PRC_2&quot;)
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;ACTA2+PRC_3&quot;] &lt;- paste0(&quot;ACTA2&quot;, expression(&quot;\U207A&quot;),&quot;PRC_3&quot;)
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;ACTA2+PRC_4&quot;] &lt;- paste0(&quot;ACTA2&quot;, expression(&quot;\U207A&quot;),&quot;PRC_4&quot;)
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;ACTA2+PRC_5&quot;] &lt;- paste0(&quot;ACTA2&quot;, expression(&quot;\U207A&quot;),&quot;PRC_5&quot;)
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;FDC_6&quot;] &lt;- &quot;FDC&quot;
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;PI16+RC_10&quot;] &lt;- paste0(&quot;PI16&quot;, expression(&quot;\U207A&quot;),&quot;RC_1&quot;)
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;PI16+RC_11&quot;] &lt;- paste0(&quot;PI16&quot;, expression(&quot;\U207A&quot;),&quot;RC_2&quot;)
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;PI16+RC_12&quot;] &lt;- paste0(&quot;PI16&quot;, expression(&quot;\U207A&quot;),&quot;RC_3&quot;)
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;TRC_7&quot;] &lt;- &quot;TRC_1&quot;
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;TRC_8&quot;] &lt;- &quot;TRC_2&quot;
Tons_FRC_data@meta.data$clusterLabel[Tons_FRC_data@meta.data$clusterLabel == &quot;TRC_9&quot;] &lt;- &quot;TRC_3&quot;

colDataset &lt;- cols[3:15]
names(colDataset) &lt;- unique(Tons_FRC_data$clusterLabel)

DimPlot(Tons_FRC_data, reduction = &quot;umap&quot;, group.by = &quot;clusterLabel&quot;,cols=colDataset)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;) + ggtitle(&quot;Tonsilar FRCs (De Martin et al 2023)&quot;)</code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/umap%20Tonsilar%20FRC-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="merge-nsclc-trc-and-prc-with-tonsilar-frcs"
class="section level3">
<h3>Merge NSCLC TRC and PRC with Tonsilar FRCs</h3>
<pre class="r"><code>NCLS_FRCS$Disease_short &lt;-rep(&quot;NSCLC&quot;,nrow(NCLS_FRCS@meta.data))
Tons_FRC_data$Disease_short &lt;-rep(&quot;Tonsil&quot;,nrow(Tons_FRC_data@meta.data))
colnames(Tons_FRC_data@meta.data)[names(Tons_FRC_data@meta.data) == &#39;clusterLabel&#39;] &lt;- &#39;cell_type&#39;

data_merge &lt;- merge(NCLS_FRCS, y = c(Tons_FRC_data),
             add.cell.ids = c(&quot;NCLS_FRCS&quot;,&quot;Tons_FRC_data&quot;),
             project = &quot;merge_nsclc_tonsils&quot;)

#Preprocessing
resolution &lt;- c(0.1, 0.25, 0.4, 0.6,0.8, 1.)
data_merge  &lt;- preprocessing(data_merge,resolution)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33594
Number of edges: 1098322

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9641
Number of communities: 6
Elapsed time: 6 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33594
Number of edges: 1098322

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9342
Number of communities: 12
Elapsed time: 7 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33594
Number of edges: 1098322

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9153
Number of communities: 13
Elapsed time: 6 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33594
Number of edges: 1098322

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8977
Number of communities: 17
Elapsed time: 6 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33594
Number of edges: 1098322

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8821
Number of communities: 21
Elapsed time: 5 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 33594
Number of edges: 1098322

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8698
Number of communities: 27
Elapsed time: 5 seconds</code></pre>
</div>
<div
id="integrate-data-to-correct-for-batch-effects-due-to-different-tissues-via-seurat"
class="section level3">
<h3><strong>Integrate data to correct for batch effects due to different
tissues via seurat</strong></h3>
</div>
<div id="step-1" class="section level3">
<h3>Step 1</h3>
<pre class="r"><code>obj.list &lt;-SplitObject(data_merge, split.by = &#39;cell_type&#39;)
#For each object in list we see to run normalization and identify highly variable features
for (i in 1:length(obj.list)){
  #Normalization
  obj.list[[i]] &lt;- NormalizeData(obj.list[[i]], normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000) 
  #Find high variable genes
  obj.list[[i]] &lt;- FindVariableFeatures(obj.list[[i]], selection.method = &quot;vst&quot;, nfeatures = 2000)
}</code></pre>
</div>
<div id="step-2" class="section level3">
<h3>Step 2</h3>
<pre class="r"><code>#select features that are repeatedly variable across datasets for integration
features &lt;- SelectIntegrationFeatures(object.list = obj.list)
#Find anchors to integrate the data across different patients (Canonical correlation analysis)
anchors &lt;- FindIntegrationAnchors(object.list = obj.list, anchor.features = features)
# Create an &#39;integrated&#39; data assay
seurat_integrated &lt;- IntegrateData(anchorset = anchors)</code></pre>
</div>
<div id="step-3" class="section level3">
<h3>Step 3</h3>
<pre class="r"><code># We run a single integrated analysis on all cells!
DefaultAssay(seurat_integrated) &lt;- &quot;integrated&quot;

# Run the standard workflow for visualization and clustering
seurat_integrated &lt;- ScaleData(seurat_integrated, verbose = FALSE)
seurat_integrated &lt;- RunPCA(object = seurat_integrated, npcs = 30, verbose = FALSE,seed.use = 8734) 
seurat_integrated &lt;- RunTSNE(object = seurat_integrated, reduction = &quot;pca&quot;, dims = 1:20, seed.use = 8734)
seurat_integrated&lt;- RunUMAP(object = seurat_integrated, reduction = &quot;pca&quot;, dims = 1:20, seed.use = 8734)
seurat_integrated &lt;- FindNeighbors(object = seurat_integrated, reduction = &quot;pca&quot;, dims = 1:20, seed.use = 8734)
#Clustering 
resolution &lt;- c(0.1, 0.25, 0.4, 0.6,0.8, 1.,1.2,1.4,1.8)
for(k in 1:length(resolution)){
  seurat_integrated &lt;- FindClusters(object = seurat_integrated, resolution = resolution[k], random.seed = 8734)
}</code></pre>
</div>
<div
id="apply-divide-and-conquer-mds-algorithm-proposed-by-delicado-p.-and-c.-pachón-garcía-2021-for-fast-mds-computation-due-to-large-dataset-size"
class="section level3">
<h3>Apply Divide and conquer MDS algorithm proposed by Delicado P. and
C. Pachón-García (2021) for fast MDS computation due to large dataset
size</h3>
<pre class="r"><code># celltype similarity wth MDS
DefaultAssay(seurat_integrated) &lt;-&#39;integrated&#39;

#Divide-andconquer MDS proposed by Delicado P. and C. Pachón-García (2021)
#MDS computation
mds &lt;- divide_conquer_mds(x = t(GetAssayData(seurat_integrated, slot = &#39;scale.data&#39;)), l = 200, c_points = 5 * 2, r = 2, n_cores = 1)$points
colnames(mds) &lt;- paste0(&quot;MDSDIVCONQ_&quot;, 1:2)

# Store MDS representation as a custom dimensional reduction field
seurat_integrated[[&#39;mds_div_conq&#39;]] &lt;- CreateDimReducObject(embeddings = mds, key = &#39;MDSDIVCONQ_&#39;, assay = DefaultAssay(seurat_integrated))

# Save seurat object for later use
#saveRDS(seurat_integrated,paste0(basedir,&quot;/data/Human/Tonsil_Ccl19_TRC_PRC_final_mds_div_conq.rds&quot;))</code></pre>
</div>
<div id="multidimensional-scaling-mds-plot" class="section level3">
<h3>Multidimensional scaling (MDS) plot</h3>
</div>
<div id="read-integrated-object-with-mds-representation"
class="section level3">
<h3>Read integrated object with MDS representation</h3>
<pre class="r"><code>seurat_integrated &lt;- readRDS(paste0(basedir,&quot;/data/Human/Tonsil_Ccl19_TRC_PRC_final_mds_conq.rds&quot;))</code></pre>
</div>
<div id="visualize-mds-plot-figure-4d" class="section level3">
<h3>Visualize MDS plot (Figure 4D)</h3>
<pre class="r"><code>mds_tx_condition &lt;- seurat_integrated@reductions$mds_div_conq@cell.embeddings %&gt;% 
as.data.frame() %&gt;% cbind(tx = seurat_integrated@meta.data$Disease_short)

mds_tx_celltype &lt;- seurat_integrated@reductions$mds_div_conq@cell.embeddings %&gt;% 
as.data.frame() %&gt;% cbind(tx = seurat_integrated@meta.data$cell_type)

mds_tx_TOTAL &lt;- merge(mds_tx_condition, mds_tx_celltype, by=c(&quot;MDSDIVCONQ_1&quot;, &quot;MDSDIVCONQ_2&quot;), all.x=T, all.y=T)
colnames(mds_tx_TOTAL) &lt;-c(&quot;MDS_1&quot;, &quot;MDS_2&quot;, &quot;Condition&quot;,&quot;Celltype&quot;)

#Color palette
colDataset &lt;- cols[1:15]
names(colDataset) &lt;- unique(seurat_integrated$cell_type)

# Use mean gaussian kernel
mds_tx_TOTAL_gk &lt;- mds_tx_TOTAL %&gt;% 
  group_by(Celltype,Condition) %&gt;% 
  mutate(count_mds1 = mean(GK(MDS_1))) %&gt;%
  mutate(count_mds2 = mean(GK(MDS_2)))

ggplot(mds_tx_TOTAL_gk, aes(x=count_mds1, y=count_mds2, color=Celltype, shape = Condition)) + geom_point(stroke = 1.5) + ylab(&quot;MDS2&quot;) + xlab(&quot;MDS1&quot;) + coord_cartesian(xlim = c(0, max(mds_tx_TOTAL_gk$count_mds1,mds_tx_TOTAL_gk$count_mds2)), ylim = c(0, max(mds_tx_TOTAL_gk$count_mds1,mds_tx_TOTAL_gk$count_mds2)) ) +
scale_color_manual(values=colDataset) + scale_shape_manual(values = c(2, 3)) + 
  theme(aspect.ratio = 2,axis.line = element_line(colour = &quot;black&quot;),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank(),
    panel.background = element_blank(),
    axis.text.y = element_text(angle = 0, vjust = 0.5,colour = &quot;black&quot;,size = 10),
    axis.text.x = element_text(angle = 0, vjust = 0.5,colour = &quot;black&quot;,size = 10)) </code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/MDS%20visualization-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="read-nsclc-til-data" class="section level3">
<h3>Read NSCLC TIL data</h3>
<pre class="r"><code>NSCLS_TIL_data &lt;- readRDS(paste0(basedir,&quot;/data/Human/NSCLC_TILs.rds&quot;))</code></pre>
</div>
<div id="merge-nscls-ccl19-frcs-and-nscls-tils" class="section level3">
<h3>Merge NSCLS CCL19⁺ FRCs and NSCLS TILs</h3>
<pre class="r"><code>same_columns &lt;- intersect(colnames(NSCLS_TIL_data@meta.data),colnames(NCLS_FRCS@meta.data))

NSCLS_TIL_data@meta.data &lt;-NSCLS_TIL_data@meta.data[,same_columns]
NCLS_FRCS@meta.data &lt;-NCLS_FRCS@meta.data[,same_columns]

merged_data&lt;- merge(NSCLS_TIL_data, y = c(NCLS_FRCS),
             add.cell.ids = c(&#39;NSCLS_TIL_data&#39;,&#39;NCLS_FRCS&#39;),
             project = &quot;NSCLC_FRC_TIL&quot;)

resolution &lt;- c(0.1, 0.25, 0.4, 0.6, 0.8, 1.,1.2,1.4,1.6,2.)
merged_data  &lt;- preprocessing(merged_data,resolution)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9692
Number of communities: 7
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9416
Number of communities: 11
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9236
Number of communities: 13
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9030
Number of communities: 15
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8869
Number of communities: 20
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8738
Number of communities: 21
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8610
Number of communities: 22
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8499
Number of communities: 24
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8399
Number of communities: 28
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 11699
Number of edges: 402688

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8217
Number of communities: 29
Elapsed time: 1 seconds</code></pre>
</div>
<div
id="frc-immune-cell-interactions-in-nsclc-via-cellchat-suoqin-jin-et-al.-2021"
class="section level3">
<h3><strong>FRC-immune cell interactions in NSCLC via Cellchat (Suoqin
Jin et al., 2021)</strong></h3>
</div>
<div id="convert-seurat-object-to-cellchat-object"
class="section level3">
<h3>Convert seurat object to cellchat object</h3>
<pre class="r"><code>#Define color palet
palet &lt;- c(&quot;#5050FFFF&quot;, &quot;#CE3D32FF&quot;, &quot;#4DAF4A&quot;,&quot;#FB9A99&quot;,&quot;#377EB8&quot;,&quot;#A65628&quot;,&quot;#222F75&quot;)
names(palet) &lt;- c( &quot;CAF2/TRC&quot;,&quot;CAF1/PRC&quot;, paste0(&quot;CD4&quot;, &quot;\u207A &quot;, &quot;T cells&quot;), paste0(&quot;CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;), &quot;B cells&quot;, &quot;Regulatory T cells&quot;,paste0(&quot;Cycling CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;))

# Interactome analysis
cellchat &lt;- Cellchat_Analysis(merged_data)</code></pre>
<pre><code>[1] &quot;Create a CellChat object from a data matrix&quot;
Set cell identities for the new CellChat object 
The cell groups used for CellChat analysis are  B cells CAF1/PRC CAF2/TRC CD4⁺ T cells CD8⁺ T cells Cycling CD8⁺ T cells Regulatory T cells </code></pre>
<pre class="r"><code>cellchat &lt;-CellChatDownstreamAnalysis(cellchat,&quot;human&quot;,thresh = 0.05)</code></pre>
<pre><code>triMean is used for calculating the average gene expression per cell group. 
[1] &quot;&gt;&gt;&gt; Run CellChat on sc/snRNA-seq data &lt;&lt;&lt; [2025-10-01 17:47:36.645817]&quot;
[1] &quot;&gt;&gt;&gt; CellChat inference is done. Parameter values are stored in `object@options$parameter` &lt;&lt;&lt; [2025-10-01 17:49:04.240146]&quot;</code></pre>
</div>
<div id="interactome-analysis-figure-4e" class="section level3">
<h3>Interactome analysis (Figure 4E)</h3>
<pre class="r"><code>gg &lt;- netAnalysis_signalingRole_scatter(cellchat,color.use = palet)
gg &lt;- gg + ggtitle(&quot;Interactome analysis (Cellchat)&quot;)
gg</code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/interactome%20analysis-1.png" width="672" style="display: block; margin: auto;" />
### Outgoing signaling patterns</p>
<pre class="r"><code>selectK(cellchat, pattern = &quot;outgoing&quot;)</code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/outgoing%20signaling-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="incoming-signaling-patterns" class="section level3">
<h3>Incoming signaling patterns</h3>
<pre class="r"><code>selectK(cellchat, pattern = &quot;incoming&quot;)</code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/incoming%20signaling-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="incoming-and-outgoing-patterns" class="section level3">
<h3>Incoming and outgoing patterns</h3>
<pre class="r"><code>cellchat &lt;- identifyCommunicationPatterns(cellchat, pattern = &quot;outgoing&quot;, k = 5, color.use = palet)
cellchat &lt;- identifyCommunicationPatterns(cellchat, pattern = &quot;incoming&quot;, k = 6, color.use = palet)</code></pre>
</div>
<div
id="joint-dotplot-for-highlighting-incoming-and-outcoming-communication-patterns-explicitly-for-pathways-with-active-signaling-between-frcs-and-immune-cells-figure-4f"
class="section level3">
<h3>Joint dotplot for highlighting incoming and outcoming communication
patterns explicitly for pathways with active signaling between FRCs and
immune cells (Figure 4F)</h3>
<pre class="r"><code>order_list &lt;-c(&quot;CAF2/TRC&quot;,&quot;CAF1/PRC&quot;, 
               paste0(&quot;CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Cycling CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;CD4&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),&quot;B cells&quot;, &quot;Regulatory T cells&quot;)

pathways&lt;- cellchat@netP$pathways

gg &lt;-comAnalysis_joint_dot(cellchat,color.use = palet,font.size = 8,pathways = pathways, order_list = order_list)$gg
shared_signaling &lt;- comAnalysis_joint_dot(cellchat,color.use = palet,font.size = 8,pathways = pathways, order_list = order_list)$new_df_list
gg</code></pre>
<p><img src="figure/comparison_CCL19_FRC_NSCLC_tonsil.Rmd/join%20dotplot%20FRC%20immune-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="session-info" class="section level3">
<h3>Session info</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] doParallel_1.0.17   iterators_1.0.14    foreach_1.5.2      
 [4] bigmds_3.0.0        ggsci_3.0.0         harmony_1.2.0      
 [7] Rcpp_1.0.11         CellChat_1.6.1      igraph_1.5.0.1     
[10] dplyr_1.1.2         NMF_0.26            Biobase_2.60.0     
[13] BiocGenerics_0.46.0 cluster_2.1.4       rngtools_1.5.2     
[16] registry_0.5-1      dittoSeq_1.12.1     ggplot2_3.4.2      
[19] SeuratObject_5.1.0  Seurat_4.3.0.1      purrr_1.0.1        
[22] here_1.0.1          magrittr_2.0.3      circlize_0.4.15    
[25] tidyr_1.3.0         tibble_3.2.1        workflowr_1.7.1    

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21            splines_4.3.1              
  [3] later_1.3.1                 bitops_1.0-7               
  [5] polyclip_1.10-4             ggnetwork_0.5.12           
  [7] lifecycle_1.0.3             rstatix_0.7.2              
  [9] rprojroot_2.0.3             globals_0.16.2             
 [11] processx_3.8.2              lattice_0.21-8             
 [13] MASS_7.3-60                 backports_1.4.1            
 [15] plotly_4.10.2               sass_0.4.7                 
 [17] rmarkdown_2.23              jquerylib_0.1.4            
 [19] yaml_2.3.7                  httpuv_1.6.11              
 [21] sctransform_0.4.1           spam_2.10-0                
 [23] sp_2.0-0                    spatstat.sparse_3.0-2      
 [25] reticulate_1.36.1           cowplot_1.1.1              
 [27] pbapply_1.7-2               RColorBrewer_1.1-3         
 [29] abind_1.4-5                 zlibbioc_1.46.0            
 [31] Rtsne_0.16                  GenomicRanges_1.52.0       
 [33] RCurl_1.98-1.12             pracma_2.4.4               
 [35] git2r_0.33.0                GenomeInfoDbData_1.2.10    
 [37] IRanges_2.34.1              S4Vectors_0.38.1           
 [39] svd_0.5.5                   ggrepel_0.9.3              
 [41] irlba_2.3.5.1               listenv_0.9.0              
 [43] spatstat.utils_3.1-0        pheatmap_1.0.12            
 [45] RSpectra_0.16-1             goftest_1.2-3              
 [47] spatstat.random_3.1-5       fitdistrplus_1.1-11        
 [49] parallelly_1.36.0           svglite_2.1.1              
 [51] leiden_0.4.3                codetools_0.2-19           
 [53] DelayedArray_0.28.0         tidyselect_1.2.0           
 [55] shape_1.4.6                 farver_2.1.1               
 [57] matrixStats_1.0.0           stats4_4.3.1               
 [59] spatstat.explore_3.2-1      jsonlite_1.8.7             
 [61] GetoptLong_1.0.5            BiocNeighbors_1.18.0       
 [63] ellipsis_0.3.2              progressr_0.13.0           
 [65] ggalluvial_0.12.5           ggridges_0.5.4             
 [67] survival_3.5-5              systemfonts_1.0.4          
 [69] tools_4.3.1                 ragg_1.2.5                 
 [71] sna_2.7-1                   ica_1.0-3                  
 [73] glue_1.6.2                  gridExtra_2.3              
 [75] SparseArray_1.2.4           xfun_0.39                  
 [77] MatrixGenerics_1.12.3       GenomeInfoDb_1.36.1        
 [79] withr_2.5.0                 BiocManager_1.30.21.1      
 [81] fastmap_1.1.1               fansi_1.0.4                
 [83] callr_3.7.3                 digest_0.6.33              
 [85] R6_2.5.1                    mime_0.12                  
 [87] textshaping_0.3.6           colorspace_2.1-0           
 [89] Cairo_1.6-1                 scattermore_1.2            
 [91] tensor_1.5                  spatstat.data_3.0-1        
 [93] utf8_1.2.3                  generics_0.1.3             
 [95] data.table_1.14.8           FNN_1.1.3.2                
 [97] httr_1.4.6                  htmlwidgets_1.6.2          
 [99] S4Arrays_1.2.1              whisker_0.4.1              
[101] uwot_0.1.16                 pkgconfig_2.0.3            
[103] gtable_0.3.3                ComplexHeatmap_2.16.0      
[105] lmtest_0.9-40               SingleCellExperiment_1.22.0
[107] XVector_0.40.0              htmltools_0.5.5            
[109] carData_3.0-5               dotCall64_1.1-1            
[111] clue_0.3-64                 scales_1.2.1               
[113] png_0.1-8                   knitr_1.43                 
[115] rstudioapi_0.15.0           rjson_0.2.21               
[117] reshape2_1.4.4              coda_0.19-4                
[119] statnet.common_4.9.0        nlme_3.1-162               
[121] cachem_1.0.8                zoo_1.8-12                 
[123] GlobalOptions_0.1.2         stringr_1.5.0              
[125] KernSmooth_2.23-22          miniUI_0.1.1.1             
[127] pillar_1.9.0                grid_4.3.1                 
[129] vctrs_0.6.3                 RANN_2.6.1                 
[131] ggpubr_0.6.0                promises_1.2.0.1           
[133] car_3.1-2                   xtable_1.8-4               
[135] evaluate_0.21               cli_3.6.1                  
[137] compiler_4.3.1              rlang_1.1.1                
[139] crayon_1.5.2                ggsignif_0.6.4             
[141] future.apply_1.11.0         labeling_0.4.2             
[143] ps_1.7.5                    forcats_1.0.0              
[145] getPass_0.2-4               plyr_1.8.8                 
[147] fs_1.6.3                    stringi_1.7.12             
[149] network_1.18.1              BiocParallel_1.34.2        
[151] viridisLite_0.4.2           deldir_1.0-9               
[153] gridBase_0.4-7              munsell_0.5.0              
[155] lazyeval_0.2.2              spatstat.geom_3.2-4        
[157] Matrix_1.6-4                patchwork_1.1.2            
[159] future_1.33.0               shiny_1.7.4.1              
[161] highr_0.10                  SummarizedExperiment_1.30.2
[163] ROCR_1.0-11                 broom_1.0.5                
[165] bslib_0.5.0                </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Wed Oct  1 17:52:45 2025&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 15.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] doParallel_1.0.17   iterators_1.0.14    foreach_1.5.2      
 [4] bigmds_3.0.0        ggsci_3.0.0         harmony_1.2.0      
 [7] Rcpp_1.0.11         CellChat_1.6.1      igraph_1.5.0.1     
[10] dplyr_1.1.2         NMF_0.26            Biobase_2.60.0     
[13] BiocGenerics_0.46.0 cluster_2.1.4       rngtools_1.5.2     
[16] registry_0.5-1      dittoSeq_1.12.1     ggplot2_3.4.2      
[19] SeuratObject_5.1.0  Seurat_4.3.0.1      purrr_1.0.1        
[22] here_1.0.1          magrittr_2.0.3      circlize_0.4.15    
[25] tidyr_1.3.0         tibble_3.2.1        workflowr_1.7.1    

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21            splines_4.3.1              
  [3] later_1.3.1                 bitops_1.0-7               
  [5] polyclip_1.10-4             ggnetwork_0.5.12           
  [7] lifecycle_1.0.3             rstatix_0.7.2              
  [9] rprojroot_2.0.3             globals_0.16.2             
 [11] processx_3.8.2              lattice_0.21-8             
 [13] MASS_7.3-60                 backports_1.4.1            
 [15] plotly_4.10.2               sass_0.4.7                 
 [17] rmarkdown_2.23              jquerylib_0.1.4            
 [19] yaml_2.3.7                  httpuv_1.6.11              
 [21] sctransform_0.4.1           spam_2.10-0                
 [23] sp_2.0-0                    spatstat.sparse_3.0-2      
 [25] reticulate_1.36.1           cowplot_1.1.1              
 [27] pbapply_1.7-2               RColorBrewer_1.1-3         
 [29] abind_1.4-5                 zlibbioc_1.46.0            
 [31] Rtsne_0.16                  GenomicRanges_1.52.0       
 [33] RCurl_1.98-1.12             pracma_2.4.4               
 [35] git2r_0.33.0                GenomeInfoDbData_1.2.10    
 [37] IRanges_2.34.1              S4Vectors_0.38.1           
 [39] svd_0.5.5                   ggrepel_0.9.3              
 [41] irlba_2.3.5.1               listenv_0.9.0              
 [43] spatstat.utils_3.1-0        pheatmap_1.0.12            
 [45] RSpectra_0.16-1             goftest_1.2-3              
 [47] spatstat.random_3.1-5       fitdistrplus_1.1-11        
 [49] parallelly_1.36.0           svglite_2.1.1              
 [51] leiden_0.4.3                codetools_0.2-19           
 [53] DelayedArray_0.28.0         tidyselect_1.2.0           
 [55] shape_1.4.6                 farver_2.1.1               
 [57] matrixStats_1.0.0           stats4_4.3.1               
 [59] spatstat.explore_3.2-1      jsonlite_1.8.7             
 [61] GetoptLong_1.0.5            BiocNeighbors_1.18.0       
 [63] ellipsis_0.3.2              progressr_0.13.0           
 [65] ggalluvial_0.12.5           ggridges_0.5.4             
 [67] survival_3.5-5              systemfonts_1.0.4          
 [69] tools_4.3.1                 ragg_1.2.5                 
 [71] sna_2.7-1                   ica_1.0-3                  
 [73] glue_1.6.2                  gridExtra_2.3              
 [75] SparseArray_1.2.4           xfun_0.39                  
 [77] MatrixGenerics_1.12.3       GenomeInfoDb_1.36.1        
 [79] withr_2.5.0                 BiocManager_1.30.21.1      
 [81] fastmap_1.1.1               fansi_1.0.4                
 [83] callr_3.7.3                 digest_0.6.33              
 [85] R6_2.5.1                    mime_0.12                  
 [87] textshaping_0.3.6           colorspace_2.1-0           
 [89] Cairo_1.6-1                 scattermore_1.2            
 [91] tensor_1.5                  spatstat.data_3.0-1        
 [93] utf8_1.2.3                  generics_0.1.3             
 [95] data.table_1.14.8           FNN_1.1.3.2                
 [97] httr_1.4.6                  htmlwidgets_1.6.2          
 [99] S4Arrays_1.2.1              whisker_0.4.1              
[101] uwot_0.1.16                 pkgconfig_2.0.3            
[103] gtable_0.3.3                ComplexHeatmap_2.16.0      
[105] lmtest_0.9-40               SingleCellExperiment_1.22.0
[107] XVector_0.40.0              htmltools_0.5.5            
[109] carData_3.0-5               dotCall64_1.1-1            
[111] clue_0.3-64                 scales_1.2.1               
[113] png_0.1-8                   knitr_1.43                 
[115] rstudioapi_0.15.0           rjson_0.2.21               
[117] reshape2_1.4.4              coda_0.19-4                
[119] statnet.common_4.9.0        nlme_3.1-162               
[121] cachem_1.0.8                zoo_1.8-12                 
[123] GlobalOptions_0.1.2         stringr_1.5.0              
[125] KernSmooth_2.23-22          miniUI_0.1.1.1             
[127] pillar_1.9.0                grid_4.3.1                 
[129] vctrs_0.6.3                 RANN_2.6.1                 
[131] ggpubr_0.6.0                promises_1.2.0.1           
[133] car_3.1-2                   xtable_1.8-4               
[135] evaluate_0.21               cli_3.6.1                  
[137] compiler_4.3.1              rlang_1.1.1                
[139] crayon_1.5.2                ggsignif_0.6.4             
[141] future.apply_1.11.0         labeling_0.4.2             
[143] ps_1.7.5                    forcats_1.0.0              
[145] getPass_0.2-4               plyr_1.8.8                 
[147] fs_1.6.3                    stringi_1.7.12             
[149] network_1.18.1              BiocParallel_1.34.2        
[151] viridisLite_0.4.2           deldir_1.0-9               
[153] gridBase_0.4-7              munsell_0.5.0              
[155] lazyeval_0.2.2              spatstat.geom_3.2-4        
[157] Matrix_1.6-4                patchwork_1.1.2            
[159] future_1.33.0               shiny_1.7.4.1              
[161] highr_0.10                  SummarizedExperiment_1.30.2
[163] ROCR_1.0-11                 broom_1.0.5                
[165] bslib_0.5.0                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
