<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chrysa Papadopoulou" />


<title>Comparison of naive CCL19-EYFP cells vs mCOV-FIt31-g33 CCL19-EYFP cells</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CCL19_FRCs_lung_cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Comparison of naive CCL19-EYFP cells vs
mCOV-FIt31-g33 CCL19-EYFP cells</h1>
<h4 class="author">Chrysa Papadopoulou</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CCL19_FRCs_lung_cancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240808code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240808)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240808code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240808)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomPchryssaCCL19FRCslungcancertree05125c298c1d49c3497c6d5447c282ead1c9eed9targetblank05125c2a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/tree/05125c298c1d49c3497c6d5447c282ead1c9eed9" target="_blank">05125c2</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomPchryssaCCL19FRCslungcancertree05125c298c1d49c3497c6d5447c282ead1c9eed9targetblank05125c2a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/tree/05125c298c1d49c3497c6d5447c282ead1c9eed9" target="_blank">05125c2</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    data/Final_submission/
    Ignored:    data/Human/
    Ignored:    data/Mouse/
    Ignored:    data/Public/
    Ignored:    output/GSEA_AdvFB_SULF1/
    Ignored:    output/GSEA_AdvFB_TLS/
    Ignored:    output/GSEA_CCR7_T/
    Ignored:    output/GSEA_CD8_T/
    Ignored:    output/GSEA_CYCL_T/
    Ignored:    output/GSEA_EXH_T/
    Ignored:    output/GSEA_SMC_PRC/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/.h5seurat
    Untracked:  analysis/Compare_tumors.Rmd
    Untracked:  analysis/NSCLC_PDAC_CAFs.Rmd
    Untracked:  analysis/Seurat_to_SCE.Rmd
    Untracked:  analysis/compression.Rmd
    Untracked:  analysis/index_hidden.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/mcov_R.Rmd</code>) and HTML
(<code>docs/mcov_R.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/05125c298c1d49c3497c6d5447c282ead1c9eed9/analysis/mcov_R.Rmd" target="_blank">05125c2</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-09-23
</td>
<td>
CCL19-EYFP cells vs mCOV-FIt31-g33 CCL19-EYFP cells
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>suppressPackageStartupMessages({
  library(here)
  library(purrr)
  library(dplyr)
  library(stringr)
  library(patchwork)
  library(Seurat)
  library(Matrix)
  library(gridExtra)
  library(gsubfn)
  library(ggsci)
  library(biomaRt)
  library(tidyverse)
  library(msigdbr)
  library(stats)
  library(clusterProfiler)
  library(dict)
  library(openxlsx)
  library(DOSE)
  library(enrichplot)
})</code></pre>
</div>
<div id="set-directory" class="section level3">
<h3>Set directory</h3>
<pre class="r"><code>basedir &lt;- here()</code></pre>
</div>
<div
id="read-ccl19-eyfp-cell-data-from-naïve-lungs-and-excised-llc-gp33-tumors-on-day-23"
class="section level3">
<h3>Read CCL19-EYFP⁺ cell data from naïve lungs and excised LLC-gp33
tumors on day 23</h3>
<pre class="r"><code>CCL19_EYFP &lt;- readRDS(paste0(basedir,&quot;/data/Mouse/CCL19_EYFP_nonmCOV.rds&quot;))</code></pre>
</div>
<div id="read-ccl19-eyfp-mcov-fit31-g33-cell-data"
class="section level3">
<h3>Read CCL19-EYFP⁺ mCOV-FIt31-g33 cell data</h3>
<pre class="r"><code>CCL19_EYFP_mCOV &lt;- readRDS(paste0(basedir,&quot;/data/Mouse/mCOV.rds&quot;))</code></pre>
</div>
<div id="feature-plots" class="section level3 tabset">
<h3 class="tabset">Feature plots</h3>
<div id="cxcl13-llc-naive" class="section level4">
<h4>Cxcl13 LLC naive</h4>
<pre class="r"><code>FeaturePlot(CCL19_EYFP, reduction = &quot;umap&quot;,
          features = get_full_gene_name(&#39;Cxcl13&#39;,CCL19_EYFP),raster=FALSE,
          cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),min.cutoff = 0, max.cutoff = 6) + ggtitle(paste0(&quot;Ccl19-EYFP&quot;, &quot;\U207A &quot;, &quot;cells (Naive)&quot;))</code></pre>
<p><img src="figure/mcov_R.Rmd/CXCL13%20LLC%20naive-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cxcl13-mcov-fit31-g33" class="section level4">
<h4>Cxcl13 mCOV-FIt31-g33</h4>
<pre class="r"><code>FeaturePlot(CCL19_EYFP_mCOV, reduction = &quot;umap&quot;, 
          features = get_full_gene_name(&#39;Cxcl13&#39;,CCL19_EYFP_mCOV),raster=FALSE,
          cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;)) + ggtitle(paste0(&quot;Ccl19-EYFP&quot;, &quot;\U207A &quot;, &quot;cells (mCOV-FIt31-g33)&quot;))</code></pre>
<p><img src="figure/mcov_R.Rmd/CXCL13%20mCOV-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cr2-mcov-fit31-g33" class="section level4">
<h4>Cr2 mCOV-FIt31-g33</h4>
<pre class="r"><code>FeaturePlot(CCL19_EYFP_mCOV, reduction = &quot;umap&quot;, 
          features = get_full_gene_name(&#39;Cr2&#39;,CCL19_EYFP_mCOV),raster=FALSE,
          cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;)) + ggtitle(paste0(&quot;Ccl19-EYFP&quot;, &quot;\U207A &quot;, &quot;cells (mCOV-FIt31-g33)&quot;))</code></pre>
<p><img src="figure/mcov_R.Rmd/CR2%20mCOV-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><strong>Let’s compare the gene expression between TRC/PRC (d23) and
their progenitor subsets (d15) to identify gene programs that may be
relevant for their function in supporting the T cell niches</strong></p>
<pre class="r"><code>#Set new annotation: Change Smoc1 AdvFB to Cd34 AdvFB
CCL19_EYFP@meta.data$annot[CCL19_EYFP@meta.data$annot == paste0(&quot;Smoc1&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;) ] &lt;- paste0(&quot;Cd34&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;)

progenitors_d15 &lt;- subset(CCL19_EYFP, TimePoint == &quot;d15&quot; &amp; annot %in% c(paste0(&quot;Cd34&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),&quot;SMC/PC&quot;))

progenitors_mCOV &lt;- subset(CCL19_EYFP_mCOV, annot %in% c(paste0(&quot;Rgs5&quot;, expression(&quot;\u207A &quot;), &quot;PRC&quot;),paste0(&quot;Sulf1&quot;, expression(&quot;\u207A &quot;), &quot;TRC&quot;),&quot;TLS TRC&quot;))

data_merge &lt;- merge(progenitors_d15, y = c(progenitors_mCOV),
             add.cell.ids = c(&quot;progenitors_d15&quot;,&quot;progenitors_mCOV&quot;),
             project = &quot;progenitors_d15_mCOV&quot;)

#Preprocessing
resolution &lt;- c(0.1, 0.25, 0.4, 0.6,0.8, 1.)
data_merge  &lt;- preprocessing(data_merge,resolution)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2779
Number of edges: 94713

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9629
Number of communities: 4
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2779
Number of edges: 94713

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9246
Number of communities: 6
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2779
Number of edges: 94713

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8931
Number of communities: 10
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2779
Number of edges: 94713

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8597
Number of communities: 12
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2779
Number of edges: 94713

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8353
Number of communities: 14
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 2779
Number of edges: 94713

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8163
Number of communities: 14
Elapsed time: 0 seconds</code></pre>
</div>
</div>
<div
id="save-merged-data-progenitors-day-15-naive-and-day-23-mcov-fit31-g33"
class="section level3">
<h3>Save merged data progenitors Day 15 (naive) and Day 23
(mCOV-FIt31-g33)</h3>
<pre class="r"><code>#saveRDS(data_merge,paste0(basedir,&quot;/data/Human/progenitors_d15_mCOV.rds&quot;))</code></pre>
</div>
<div id="pathway-analysis" class="section level3 tabset">
<h3 class="tabset">Pathway analysis</h3>
<div id="cd34-advfb-day-15-vs-tls-trc-day-23-supplementary-figure-10c"
class="section level4">
<h4><strong>CD34⁺ AdvFB (Day 15) vs TLS TRC (Day 23)</strong>
(Supplementary Figure 10C)</h4>
<pre class="r"><code># Step 1 : Set output directory
subDir &lt;- &quot;GSEA_AdvFB_TLS/&quot;
saving_path &lt;- paste0(basedir,&quot;/output/&quot;)
final_dir &lt;- file.path(saving_path, subDir)
dir.create(final_dir, showWarnings = FALSE,recursive = TRUE)
map_df &lt;- ExtractMouseGeneSets(final_dir)

# Step 2: Customize parameters
httr::set_config(httr::config(ssl_verifypeer = FALSE))
organism &lt;- &quot;org.Mm.eg.db&quot;

disease_phase &lt;- &quot;d15vsd23_AdvFB_TLS&quot;
datatype &lt;- &quot;SYMBOL&quot;

advfb_tls &lt;-subset(data_merge, annot %in% c(paste0(&quot;Cd34&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),&quot;TLS TRC&quot;))

Idents(advfb_tls) &lt;- advfb_tls$annot
  
DEmarkers &lt;-FindAllMarkers(advfb_tls, only.pos=T, logfc.threshold = 0.1,
                           min.pct = 0.1)

Vec &lt;-unique(advfb_tls$annot)
EnrichParameters_TLS &lt;-customize_parameters(Vec,DEmarkers,organism,datatype,disease_phase,saving_path) </code></pre>
<pre><code>[1] &quot;Finish Enrichment_Analysis for GO Cd34⁺ AdvFB&quot;
[1] &quot;Finish Enrichment_Analysis for GO TLS TRC&quot;</code></pre>
<pre class="r"><code># Step 3: Enrichment Analysis
for (i in seq(1,length(EnrichParameters_TLS$enrichcl_list))){
  terms&lt;- EnrichParameters_TLS$enrichcl_list[[i]]
  # Filter on the most significant pathways (keep rows where p.adjust&lt;= 0.05)
  terms&lt;- terms@result[terms@result$p.adjust &lt;= 0.05,]
  population &lt;- Vec[i]
  population&lt;- gsub(&quot;/&quot;, &quot;_&quot;, population)
  write.xlsx(terms, paste0(final_dir,&quot;/&quot;,&quot;GO_Pathways_&quot;,population,&quot;.xlsx&quot;),row.names = TRUE)
}

#Step 4: Plot enriched pathways
pathways &lt;-c(&quot;cell chemotaxis&quot;, &quot;tissue remodeling&quot;,&quot;activation of immune response&quot;,
             &quot;leukocyte mediated immunity&quot;, &quot;interleukin-1 production&quot;, &quot;tumor necrosis factor production&quot;)

TLS_terms &lt;- EnrichParameters_TLS$enrichcl_list[[2]]@result

selec_pathways &lt;- TLS_terms[TLS_terms$Description %in% pathways,]

selec_pathways$Description &lt;- factor(selec_pathways$Description, levels = rev(pathways))
selec_pathways &lt;- selec_pathways[order(selec_pathways$Description), ]

ggplot(data=selec_pathways, aes(x=Description, y=qscore, fill = analysis)) + xlab(NULL) +
    geom_bar(stat=&quot;identity&quot;,position=&quot;dodge&quot;,colour = &quot;black&quot;,show.legend = FALSE, width= 0.8, size = 1 ) + coord_flip() +
    scale_y_continuous(expand = expansion(c(0,0)), limits = c(0.0, 6),breaks = c(0,2,4,6)) +
    scale_x_discrete(labels = function(x) stringr::str_wrap(x, width = 80)) +
    theme( legend.justification = &quot;top&quot;,
           plot.title = element_text(hjust = 0.5,size = 12,face=&quot;bold&quot;),axis.line = element_line(colour = &quot;black&quot;),
           panel.grid.major = element_blank(),
           panel.grid.minor = element_blank(),
           axis.text.x = element_text(angle = 0, vjust = 0.5,colour = &quot;black&quot;, size = 10),
           axis.text.y = element_text(angle = 0, vjust = 0.8,colour = &quot;black&quot;, size = 10, face=ifelse(levels(selec_pathways$Description)==&quot;tumor necrosis factor production&quot;,&quot;bold&quot;,&quot;plain&quot;)),
           axis.title.y = element_text(size = rel(2), angle = 45),
           axis.title.x = element_text(size = rel(1.5), angle = 0),
           axis.text = element_text(size = 8),
           panel.background = element_blank(), legend.position = &quot;none&quot;) +
    scale_fill_manual(values = &quot;dark gray&quot;) + ggtitle(&quot;Enriched in TLS TRC (vs. AdvFB day 15)&quot;)</code></pre>
<p><img src="figure/mcov_R.Rmd/AdvFB_TLS-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cd34-advfb-day-15-vs-sulf1-trc-day-23-supplementary-figure-10d"
class="section level4">
<h4><strong>CD34⁺ AdvFB (Day 15) vs Sulf1⁺ TRC (Day 23)</strong>
(Supplementary Figure 10D)</h4>
<pre class="r"><code># Step 1 : Set output directory
subDir &lt;- &quot;GSEA_AdvFB_SULF1/&quot;
saving_path &lt;- paste0(basedir,&quot;/output/&quot;)
final_dir &lt;- file.path(saving_path, subDir)
dir.create(final_dir, showWarnings = FALSE,recursive = TRUE)
map_df &lt;- ExtractMouseGeneSets(final_dir)

# Step 2: Customize parameters
httr::set_config(httr::config(ssl_verifypeer = FALSE))
organism &lt;- &quot;org.Mm.eg.db&quot;

disease_phase &lt;- &quot;d15vsd23_AdvFB_SULF1&quot;
datatype &lt;- &quot;SYMBOL&quot;

AdvFB_sulf1 &lt;-subset(data_merge, annot %in% c(paste0(&quot;Cd34&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),paste0(&quot;Sulf1&quot;, expression(&quot;\u207A &quot;), &quot;TRC&quot;)))

Idents(AdvFB_sulf1) &lt;- AdvFB_sulf1$annot
  
DEmarkers &lt;-FindAllMarkers(AdvFB_sulf1, only.pos=T, logfc.threshold = 0.1,
                           min.pct = 0.1)

Vec &lt;-unique(AdvFB_sulf1$annot)
EnrichParameters_Sulf1 &lt;-customize_parameters(Vec,DEmarkers,organism,datatype,disease_phase,saving_path) </code></pre>
<pre><code>[1] &quot;Finish Enrichment_Analysis for GO Cd34⁺ AdvFB&quot;
[1] &quot;Finish Enrichment_Analysis for GO Sulf1⁺ TRC&quot;</code></pre>
<pre class="r"><code># Step 3: Enrichment Analysis
for (i in seq(1,length(EnrichParameters_Sulf1$enrichcl_list))){
  terms&lt;- EnrichParameters_Sulf1$enrichcl_list[[i]]
  # Filter on the most significant pathways (keep rows where p.adjust&lt;= 0.05)
  terms&lt;- terms@result[terms@result$p.adjust &lt;= 0.05,]
  population &lt;- Vec[i]
  population&lt;- gsub(&quot;/&quot;, &quot;_&quot;, population)
  write.xlsx(terms, paste0(final_dir,&quot;/&quot;,&quot;GO_Pathways_&quot;,population,&quot;.xlsx&quot;),row.names = TRUE)
}

#Step 4: Plot enriched pathways
pathways &lt;-c(&quot;regulation of cytokine-mediated signaling pathway&quot;, &quot;lymphocyte homeostasis&quot;,&quot;actin filament bundle organization&quot;,
             &quot;positive regulation of cell-cell adhesion&quot;, &quot;regulation of T cell activation&quot;, &quot;tissue remodeling&quot;)

TRC_term_sulf1 &lt;- EnrichParameters_Sulf1$enrichcl_list[[2]]@result

selec_pathways &lt;- TRC_term_sulf1[TRC_term_sulf1$Description %in% pathways,]

selec_pathways$Description &lt;- factor(selec_pathways$Description, levels = rev(pathways))
selec_pathways &lt;- selec_pathways[order(selec_pathways$Description), ]

ggplot(data=selec_pathways, aes(x=Description, y=qscore, fill = analysis)) + xlab(NULL) +
    geom_bar(stat=&quot;identity&quot;,position=&quot;dodge&quot;,colour = &quot;black&quot;,show.legend = FALSE, width= 0.8, size = 1 ) + coord_flip() +
    scale_y_continuous(expand = expansion(c(0,0)), limits = c(0.0, 6),breaks = c(0,2,4,6)) +
    scale_x_discrete(labels = function(x) stringr::str_wrap(x, width = 80)) +
    theme( legend.justification = &quot;top&quot;,
           plot.title = element_text(hjust = 0.5,size = 12,face=&quot;bold&quot;),axis.line = element_line(colour = &quot;black&quot;),
           panel.grid.major = element_blank(),
           panel.grid.minor = element_blank(),
           axis.text.x = element_text(angle = 0, vjust = 0.5,colour = &quot;black&quot;, size = 10),
           axis.text.y = element_text(angle = 0, vjust = 0.8,colour = &quot;black&quot;, size = 10, face=ifelse(levels(selec_pathways$Description)==&quot;regulation of T cell activation&quot;,&quot;bold&quot;,&quot;plain&quot;)),
           axis.title.y = element_text(size = rel(2), angle = 45),
           axis.title.x = element_text(size = rel(1.5), angle = 0),
           axis.text = element_text(size = 8),
           panel.background = element_blank(), legend.position = &quot;none&quot;) +
    scale_fill_manual(values = &quot;dark gray&quot;) + ggtitle(paste0(&quot;Enriched in Sulf1&quot;, &quot;\U207A &quot;, &quot;TRC (vs. AdvFB day 15)&quot;)) </code></pre>
<p><img src="figure/mcov_R.Rmd/AdvFB_SULF1_TRC-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="smcpc-day-15-vs-prc-day-23-supplementary-figure-10e"
class="section level4">
<h4><strong>SMC/PC (Day 15) vs PRC (Day 23)</strong> (Supplementary
Figure 10E)</h4>
<pre class="r"><code># Step 1 : Set output directory
subDir &lt;- &quot;GSEA_SMC_PRC/&quot;
saving_path &lt;- paste0(basedir,&quot;/output/&quot;)
final_dir &lt;- file.path(saving_path, subDir)
dir.create(final_dir, showWarnings = FALSE,recursive = TRUE)
map_df &lt;- ExtractMouseGeneSets(saving_path)

# Step 2: Customize parameters
httr::set_config(httr::config(ssl_verifypeer = FALSE))
organism &lt;- &quot;org.Mm.eg.db&quot;

disease_phase &lt;- &quot;d15vsd23_SMC_PRC&quot;
datatype &lt;- &quot;SYMBOL&quot;

smc_prc &lt;-subset(data_merge, annot %in% c(&quot;SMC/PC&quot;, paste0(&quot;Rgs5&quot;, expression(&quot;\u207A &quot;), &quot;PRC&quot;)))

Idents(smc_prc) &lt;- smc_prc$annot
  
DEmarkers &lt;-FindAllMarkers(smc_prc, only.pos=T, logfc.threshold = 0.1,
                           min.pct = 0.1)

Vec &lt;-unique(smc_prc$annot)
EnrichParameters &lt;-customize_parameters(Vec,DEmarkers,organism,datatype,disease_phase,saving_path) </code></pre>
<pre><code>[1] &quot;Finish Enrichment_Analysis for GO SMC/PC&quot;
[1] &quot;Finish Enrichment_Analysis for GO Rgs5⁺ PRC&quot;</code></pre>
<pre class="r"><code># Step 3: Enrichment Analysis
for (i in seq(1,length(EnrichParameters$enrichcl_list))){
  terms&lt;- EnrichParameters$enrichcl_list[[i]]
  # Filter on the most significant pathways (keep rows where p.adjust&lt;= 0.05)
  terms&lt;- terms@result[terms@result$p.adjust &lt;= 0.05,]
  population &lt;- Vec[i]
  population&lt;- gsub(&quot;/&quot;, &quot;_&quot;, population)
  write.xlsx(terms, paste0(final_dir,&quot;/&quot;,&quot;GO_Pathways_&quot;,population,&quot;.xlsx&quot;),row.names = TRUE)
}

# Step 4: Plot enriched pathways 
pathways &lt;-c(&quot;T cell proliferation&quot;, &quot;leukocyte migration&quot;,&quot;T cell mediated immunity&quot;,
             &quot;response to type II interferon&quot;, &quot;regulation of angiogenesis&quot;, &quot;cytokine-mediated signaling pathway&quot;)

PRC_terms &lt;- EnrichParameters$enrichcl_list[[2]]@result

selec_pathways &lt;- PRC_terms[PRC_terms$Description %in% pathways,]

selec_pathways$Description &lt;- factor(selec_pathways$Description, levels = rev(pathways))
selec_pathways &lt;- selec_pathways[order(selec_pathways$Description), ]

ggplot(data=selec_pathways, aes(x=Description, y=qscore, fill = analysis)) + xlab(NULL) +
    geom_bar(stat=&quot;identity&quot;,position=&quot;dodge&quot;,colour = &quot;black&quot;,show.legend = FALSE, width= 0.8, size = 1 ) + coord_flip() +
    scale_y_continuous(expand = expansion(c(0,0)), limits = c(0.0, 15),breaks = c(0,5,10,15)) +
    scale_x_discrete(labels = function(x) stringr::str_wrap(x, width = 80)) +
    theme( legend.justification = &quot;top&quot;,
           plot.title = element_text(hjust = 0.5,size = 12,face=&quot;bold&quot;),axis.line = element_line(colour = &quot;black&quot;),
           panel.grid.major = element_blank(),
           panel.grid.minor = element_blank(),
           axis.text.x = element_text(angle = 0, vjust = 0.5,colour = &quot;black&quot;, size = 10),
           axis.text.y = element_text(angle = 0, vjust = 0.8,colour = &quot;black&quot;, size = 10, face=ifelse(levels(selec_pathways$Description)==&quot;cytokine-mediated signaling pathway&quot;,&quot;bold&quot;,&quot;plain&quot;)),
           axis.title.y = element_text(size = rel(2), angle = 45),
           axis.title.x = element_text(size = rel(1.5), angle = 0),
           axis.text = element_text(size = 8),
           panel.background = element_blank(), legend.position = &quot;none&quot;) +
    scale_fill_manual(values = &quot;dark gray&quot;) + ggtitle(&quot;Enriched in PRC (vs. SMC/PC day 15)&quot;)</code></pre>
<p><img src="figure/mcov_R.Rmd/SMC_PRC-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gene-concept-networks" class="section level3 tabset">
<h3 class="tabset">Gene-Concept Networks</h3>
<div id="enriched-in-tls-trc-vs.-advfb-day-15-supplementary-figure-10f"
class="section level4">
<h4>Enriched in TLS TRC (vs. AdvFB day 15) (Supplementary Figure
10F)</h4>
<pre class="r"><code>pathways &lt;- c(&quot;tumor necrosis factor production&quot;)

cnetplot(EnrichParameters_TLS$enrichcl_list[[2]], node_label=&quot;gene&quot;, layout = &quot;kk&quot;, showCategory = pathways,
        max.overlaps=Inf,color.params = list(gene =&quot;black&quot;,
                                             category = &quot;red&quot;,
                                             edge = TRUE),
                          cex.params = list(category_label = 0.0000001,
                                            label_gene = 0.000001, gene_label= 0.6)) + theme(legend.text=element_text(size=8))</code></pre>
<p><img src="figure/mcov_R.Rmd/network%20plot%20AdvFB_TLS-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="enriched-in-sulf1-trc-vs.-advfb-day-15-supplementary-figure-10g"
class="section level4">
<h4>Enriched in Sulf1⁺ TRC (vs. AdvFB day 15) (Supplementary Figure
10G)</h4>
<pre class="r"><code>pathways &lt;- c(&quot;regulation of T cell activation&quot;)

cnetplot(EnrichParameters_Sulf1$enrichcl_list[[2]], node_label=&quot;gene&quot;, layout = &quot;kk&quot;, showCategory = pathways,
        max.overlaps=Inf,color.params = list(gene =&quot;black&quot;,
                                             category = &quot;red&quot;,
                                             edge = TRUE),
                          cex.params = list(category_label = 0.0000001,
                                            label_gene = 0.000001, gene_label= 0.6)) + theme(legend.text=element_text(size=8))</code></pre>
<p><img src="figure/mcov_R.Rmd/network%20plot%20AdvFB_Sulf1-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-info" class="section level3">
<h3>Session info</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.6.9

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] enrichplot_1.20.0     DOSE_3.26.1           openxlsx_4.2.5.2     
 [4] dict_0.10.0           clusterProfiler_4.8.2 msigdbr_7.5.1        
 [7] lubridate_1.9.2       forcats_1.0.0         readr_2.1.4          
[10] ggplot2_3.4.2         tidyverse_2.0.0       biomaRt_2.56.1       
[13] ggsci_3.0.0           gsubfn_0.7            proto_1.0.0          
[16] gridExtra_2.3         Matrix_1.6-0          SeuratObject_4.1.3   
[19] Seurat_4.3.0.1        patchwork_1.1.2       stringr_1.5.0        
[22] dplyr_1.1.2           purrr_1.0.1           here_1.0.1           
[25] magrittr_2.0.3        circlize_0.4.15       tidyr_1.3.0          
[28] tibble_3.2.1          workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] fs_1.6.3                matrixStats_1.0.0       spatstat.sparse_3.0-2  
  [4] bitops_1.0-7            HDO.db_0.99.1           httr_1.4.6             
  [7] RColorBrewer_1.1-3      tools_4.3.1             sctransform_0.3.5      
 [10] utf8_1.2.3              R6_2.5.1                lazyeval_0.2.2         
 [13] uwot_0.1.16             withr_2.5.0             sp_2.0-0               
 [16] prettyunits_1.1.1       progressr_0.13.0        textshaping_0.3.6      
 [19] cli_3.6.1               Biobase_2.60.0          spatstat.explore_3.2-1 
 [22] scatterpie_0.2.1        labeling_0.4.2          sass_0.4.7             
 [25] spatstat.data_3.0-1     ggridges_0.5.4          pbapply_1.7-2          
 [28] systemfonts_1.0.4       yulab.utils_0.0.6       gson_0.1.0             
 [31] parallelly_1.36.0       limma_3.56.2            rstudioapi_0.15.0      
 [34] RSQLite_2.3.1           generics_0.1.3          gridGraphics_0.5-1     
 [37] shape_1.4.6             ica_1.0-3               spatstat.random_3.1-5  
 [40] zip_2.3.0               GO.db_3.17.0            fansi_1.0.4            
 [43] S4Vectors_0.38.1        abind_1.4-5             lifecycle_1.0.3        
 [46] whisker_0.4.1           yaml_2.3.7              qvalue_2.32.0          
 [49] BiocFileCache_2.8.0     Rtsne_0.16              grid_4.3.1             
 [52] blob_1.2.4              promises_1.2.0.1        crayon_1.5.2           
 [55] miniUI_0.1.1.1          lattice_0.21-8          cowplot_1.1.1          
 [58] KEGGREST_1.40.0         pillar_1.9.0            knitr_1.43             
 [61] fgsea_1.26.0            tcltk_4.3.1             future.apply_1.11.0    
 [64] codetools_0.2-19        fastmatch_1.1-4         leiden_0.4.3           
 [67] glue_1.6.2              getPass_0.2-4           downloader_0.4         
 [70] ggfun_0.1.1             data.table_1.14.8       vctrs_0.6.3            
 [73] png_0.1-8               treeio_1.24.3           org.Mm.eg.db_3.17.0    
 [76] gtable_0.3.3            cachem_1.0.8            xfun_0.39              
 [79] mime_0.12               tidygraph_1.2.3         survival_3.5-5         
 [82] ellipsis_0.3.2          fitdistrplus_1.1-11     ROCR_1.0-11            
 [85] nlme_3.1-162            ggtree_3.8.2            bit64_4.0.5            
 [88] progress_1.2.2          filelock_1.0.2          RcppAnnoy_0.0.21       
 [91] GenomeInfoDb_1.36.1     rprojroot_2.0.3         bslib_0.5.0            
 [94] irlba_2.3.5.1           KernSmooth_2.23-22      colorspace_2.1-0       
 [97] BiocGenerics_0.46.0     DBI_1.1.3               tidyselect_1.2.0       
[100] processx_3.8.2          bit_4.0.5               compiler_4.3.1         
[103] curl_5.0.1              git2r_0.33.0            xml2_1.3.5             
[106] plotly_4.10.2           shadowtext_0.1.2        scales_1.2.1           
[109] lmtest_0.9-40           callr_3.7.3             rappdirs_0.3.3         
[112] digest_0.6.33           goftest_1.2-3           spatstat.utils_3.1-0   
[115] rmarkdown_2.23          XVector_0.40.0          htmltools_0.5.5        
[118] pkgconfig_2.0.3         highr_0.10              dbplyr_2.3.3           
[121] fastmap_1.1.1           rlang_1.1.1             GlobalOptions_0.1.2    
[124] htmlwidgets_1.6.2       shiny_1.7.4.1           farver_2.1.1           
[127] jquerylib_0.1.4         zoo_1.8-12              jsonlite_1.8.7         
[130] BiocParallel_1.34.2     GOSemSim_2.26.1         RCurl_1.98-1.12        
[133] GenomeInfoDbData_1.2.10 ggplotify_0.1.1         munsell_0.5.0          
[136] Rcpp_1.0.11             ape_5.7-1               babelgene_22.9         
[139] viridis_0.6.4           reticulate_1.36.1       stringi_1.7.12         
[142] ggraph_2.1.0            zlibbioc_1.46.0         MASS_7.3-60            
[145] plyr_1.8.8              parallel_4.3.1          listenv_0.9.0          
[148] ggrepel_0.9.3           deldir_1.0-9            Biostrings_2.68.1      
[151] graphlayouts_1.0.0      splines_4.3.1           tensor_1.5             
[154] hms_1.1.3               ps_1.7.5                igraph_1.5.0.1         
[157] spatstat.geom_3.2-4     reshape2_1.4.4          stats4_4.3.1           
[160] XML_3.99-0.14           evaluate_0.21           tzdb_0.4.0             
[163] tweenr_2.0.2            httpuv_1.6.11           RANN_2.6.1             
[166] polyclip_1.10-4         future_1.33.0           scattermore_1.2        
[169] ggforce_0.4.1           xtable_1.8-4            tidytree_0.4.4         
[172] later_1.3.1             ragg_1.2.5              viridisLite_0.4.2      
[175] aplot_0.1.10            memoise_2.0.1           AnnotationDbi_1.62.2   
[178] IRanges_2.34.1          cluster_2.1.4           timechange_0.2.0       
[181] globals_0.16.2         </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Mon Sep 23 21:58:43 2024&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.6.9

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] enrichplot_1.20.0     DOSE_3.26.1           openxlsx_4.2.5.2     
 [4] dict_0.10.0           clusterProfiler_4.8.2 msigdbr_7.5.1        
 [7] lubridate_1.9.2       forcats_1.0.0         readr_2.1.4          
[10] ggplot2_3.4.2         tidyverse_2.0.0       biomaRt_2.56.1       
[13] ggsci_3.0.0           gsubfn_0.7            proto_1.0.0          
[16] gridExtra_2.3         Matrix_1.6-0          SeuratObject_4.1.3   
[19] Seurat_4.3.0.1        patchwork_1.1.2       stringr_1.5.0        
[22] dplyr_1.1.2           purrr_1.0.1           here_1.0.1           
[25] magrittr_2.0.3        circlize_0.4.15       tidyr_1.3.0          
[28] tibble_3.2.1          workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] fs_1.6.3                matrixStats_1.0.0       spatstat.sparse_3.0-2  
  [4] bitops_1.0-7            HDO.db_0.99.1           httr_1.4.6             
  [7] RColorBrewer_1.1-3      tools_4.3.1             sctransform_0.3.5      
 [10] utf8_1.2.3              R6_2.5.1                lazyeval_0.2.2         
 [13] uwot_0.1.16             withr_2.5.0             sp_2.0-0               
 [16] prettyunits_1.1.1       progressr_0.13.0        textshaping_0.3.6      
 [19] cli_3.6.1               Biobase_2.60.0          spatstat.explore_3.2-1 
 [22] scatterpie_0.2.1        labeling_0.4.2          sass_0.4.7             
 [25] spatstat.data_3.0-1     ggridges_0.5.4          pbapply_1.7-2          
 [28] systemfonts_1.0.4       yulab.utils_0.0.6       gson_0.1.0             
 [31] parallelly_1.36.0       limma_3.56.2            rstudioapi_0.15.0      
 [34] RSQLite_2.3.1           generics_0.1.3          gridGraphics_0.5-1     
 [37] shape_1.4.6             ica_1.0-3               spatstat.random_3.1-5  
 [40] zip_2.3.0               GO.db_3.17.0            fansi_1.0.4            
 [43] S4Vectors_0.38.1        abind_1.4-5             lifecycle_1.0.3        
 [46] whisker_0.4.1           yaml_2.3.7              qvalue_2.32.0          
 [49] BiocFileCache_2.8.0     Rtsne_0.16              grid_4.3.1             
 [52] blob_1.2.4              promises_1.2.0.1        crayon_1.5.2           
 [55] miniUI_0.1.1.1          lattice_0.21-8          cowplot_1.1.1          
 [58] KEGGREST_1.40.0         pillar_1.9.0            knitr_1.43             
 [61] fgsea_1.26.0            tcltk_4.3.1             future.apply_1.11.0    
 [64] codetools_0.2-19        fastmatch_1.1-4         leiden_0.4.3           
 [67] glue_1.6.2              getPass_0.2-4           downloader_0.4         
 [70] ggfun_0.1.1             data.table_1.14.8       vctrs_0.6.3            
 [73] png_0.1-8               treeio_1.24.3           org.Mm.eg.db_3.17.0    
 [76] gtable_0.3.3            cachem_1.0.8            xfun_0.39              
 [79] mime_0.12               tidygraph_1.2.3         survival_3.5-5         
 [82] ellipsis_0.3.2          fitdistrplus_1.1-11     ROCR_1.0-11            
 [85] nlme_3.1-162            ggtree_3.8.2            bit64_4.0.5            
 [88] progress_1.2.2          filelock_1.0.2          RcppAnnoy_0.0.21       
 [91] GenomeInfoDb_1.36.1     rprojroot_2.0.3         bslib_0.5.0            
 [94] irlba_2.3.5.1           KernSmooth_2.23-22      colorspace_2.1-0       
 [97] BiocGenerics_0.46.0     DBI_1.1.3               tidyselect_1.2.0       
[100] processx_3.8.2          bit_4.0.5               compiler_4.3.1         
[103] curl_5.0.1              git2r_0.33.0            xml2_1.3.5             
[106] plotly_4.10.2           shadowtext_0.1.2        scales_1.2.1           
[109] lmtest_0.9-40           callr_3.7.3             rappdirs_0.3.3         
[112] digest_0.6.33           goftest_1.2-3           spatstat.utils_3.1-0   
[115] rmarkdown_2.23          XVector_0.40.0          htmltools_0.5.5        
[118] pkgconfig_2.0.3         highr_0.10              dbplyr_2.3.3           
[121] fastmap_1.1.1           rlang_1.1.1             GlobalOptions_0.1.2    
[124] htmlwidgets_1.6.2       shiny_1.7.4.1           farver_2.1.1           
[127] jquerylib_0.1.4         zoo_1.8-12              jsonlite_1.8.7         
[130] BiocParallel_1.34.2     GOSemSim_2.26.1         RCurl_1.98-1.12        
[133] GenomeInfoDbData_1.2.10 ggplotify_0.1.1         munsell_0.5.0          
[136] Rcpp_1.0.11             ape_5.7-1               babelgene_22.9         
[139] viridis_0.6.4           reticulate_1.36.1       stringi_1.7.12         
[142] ggraph_2.1.0            zlibbioc_1.46.0         MASS_7.3-60            
[145] plyr_1.8.8              parallel_4.3.1          listenv_0.9.0          
[148] ggrepel_0.9.3           deldir_1.0-9            Biostrings_2.68.1      
[151] graphlayouts_1.0.0      splines_4.3.1           tensor_1.5             
[154] hms_1.1.3               ps_1.7.5                igraph_1.5.0.1         
[157] spatstat.geom_3.2-4     reshape2_1.4.4          stats4_4.3.1           
[160] XML_3.99-0.14           evaluate_0.21           tzdb_0.4.0             
[163] tweenr_2.0.2            httpuv_1.6.11           RANN_2.6.1             
[166] polyclip_1.10-4         future_1.33.0           scattermore_1.2        
[169] ggforce_0.4.1           xtable_1.8-4            tidytree_0.4.4         
[172] later_1.3.1             ragg_1.2.5              viridisLite_0.4.2      
[175] aplot_0.1.10            memoise_2.0.1           AnnotationDbi_1.62.2   
[178] IRanges_2.34.1          cluster_2.1.4           timechange_0.2.0       
[181] globals_0.16.2         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
