<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chrysa Papadopoulou" />


<title>Merge NSCLC FRCs and TILs</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CCL19_FRCs_lung_cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Merge NSCLC FRCs and TILs</h1>
<h4 class="author">Chrysa Papadopoulou</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-11-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CCL19_FRCs_lung_cancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240808code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240808)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240808code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240808)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomPchryssaCCL19FRCslungcancertree683e44a1a4a02e7d4631c5aee6632a4862c287eatargetblank683e44aa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/tree/683e44a1a4a02e7d4631c5aee6632a4862c287ea" target="_blank">683e44a</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomPchryssaCCL19FRCslungcancertree683e44a1a4a02e7d4631c5aee6632a4862c287eatargetblank683e44aa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/tree/683e44a1a4a02e7d4631c5aee6632a4862c287ea" target="_blank">683e44a</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    data/Final_submission/
    Ignored:    data/Human/
    Ignored:    data/Mouse/
    Ignored:    data/Public/
    Ignored:    output/GSEA_AdvFB_SULF1/
    Ignored:    output/GSEA_AdvFB_TLS/
    Ignored:    output/GSEA_CCR7_T/
    Ignored:    output/GSEA_CD8_T/
    Ignored:    output/GSEA_CYCL_T/
    Ignored:    output/GSEA_EXH_T/
    Ignored:    output/GSEA_SMC_PRC/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/.h5seurat
    Untracked:  analysis/Compare_tumors.Rmd
    Untracked:  analysis/NSCLC_PDAC_CAFs.Rmd
    Untracked:  analysis/Seurat_to_SCE.Rmd
    Untracked:  analysis/compression.Rmd
    Untracked:  analysis/index_hidden.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/merge_NSCLC_FRC_TIL.Rmd</code>)
and HTML (<code>docs/merge_NSCLC_FRC_TIL.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/683e44a1a4a02e7d4631c5aee6632a4862c287ea/analysis/merge_NSCLC_FRC_TIL.Rmd" target="_blank">683e44a</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-11-05
</td>
<td>
Correct figure ordering
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Pchryssa/CCL19_FRCs_lung_cancer/79d9795d666faf0bd87889c64bb0ce29a6d03f23/docs/merge_NSCLC_FRC_TIL.html" target="_blank">79d9795</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/7e218c6e05d46c6de593b8bf3ae221dc899ceae4/analysis/merge_NSCLC_FRC_TIL.Rmd" target="_blank">7e218c6</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-09-23
</td>
<td>
Modify figure order
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Pchryssa/CCL19_FRCs_lung_cancer/5da48e4ce280f79325a3c5716f84476b6a731a5c/docs/merge_NSCLC_FRC_TIL.html" target="_blank">5da48e4</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/c85bb484eb58ded4cefd76c1963f2086de785e48/analysis/merge_NSCLC_FRC_TIL.Rmd" target="_blank">c85bb48</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-21
</td>
<td>
Merge NSCLC TIL
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Pchryssa/CCL19_FRCs_lung_cancer/532ab316a6ea751714b197e56aa9ebc51e073194/docs/merge_NSCLC_FRC_TIL.html" target="_blank">532ab31</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/5a7acbb44878a19e3375c0f986ce5bbd5b8ce428/analysis/merge_NSCLC_FRC_TIL.Rmd" target="_blank">5a7acbb</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-21
</td>
<td>
NSCLC FRC TIL merge
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>Load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(here)
  library(purrr)
  library(stringr)
  library(patchwork)
  library(Seurat)
  library(Matrix)
  library(dittoSeq)
  library(gridExtra)
  library(CellChat)
  library(NMF)
  library(pheatmap)
})</code></pre>
</div>
<div id="infiltrating-lymphocytes-in-nsclc" class="section level2">
<h2><strong>Infiltrating lymphocytes in NSCLC</strong></h2>
<div id="set-directory" class="section level3">
<h3>Set directory</h3>
<pre class="r"><code>basedir &lt;- here()</code></pre>
</div>
</div>
<div id="merge-ccl19-frcs-and-infiltrating-lymphocytes-in-nsclc"
class="section level2">
<h2><strong>Merge CCL19⁺ FRCs and Infiltrating lymphocytes in
NSCLC</strong></h2>
<div id="read-nsclc-til-and-frc-data" class="section level3">
<h3>Read NSCLC TIL and FRC data</h3>
<pre class="r"><code>NSCLS_TIL_data &lt;-readRDS(paste0(basedir,&quot;/data/Human/NSCLC_TILs.rds&quot;))
NSCLC_CCL19_FRCs &lt;-readRDS(paste0(basedir,&quot;/data/Human/NSCLC_CCL19_FRCs_CAFs.rds&quot;))</code></pre>
</div>
<div id="merge-nscls-ccl19-frcs-and-nscls-tils" class="section level3">
<h3>Merge NSCLS CCL19⁺ FRCs and NSCLS TILs</h3>
<pre class="r"><code>same_columns &lt;- intersect(colnames(NSCLS_TIL_data@meta.data),colnames(NSCLC_CCL19_FRCs@meta.data))

NSCLS_TIL_data@meta.data &lt;-NSCLS_TIL_data@meta.data[,same_columns]
NSCLC_CCL19_FRCs@meta.data &lt;-NSCLC_CCL19_FRCs@meta.data[,same_columns]

merged_data&lt;- merge(NSCLS_TIL_data, y = c(NSCLC_CCL19_FRCs),
             add.cell.ids = c(&#39;NSCLS_TIL_data&#39;,&#39;NSCLC_CCL19_FRCs&#39;),
             project = &quot;NSCLC_FRC_TIL&quot;)

resolution &lt;- c(0.1, 0.25, 0.4, 0.6, 0.8, 1.,1.2,1.4,1.6,2.)
merged_data  &lt;- preprocessing(merged_data,resolution)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9696
Number of communities: 7
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9427
Number of communities: 11
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9254
Number of communities: 13
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9050
Number of communities: 14
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8897
Number of communities: 19
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8772
Number of communities: 22
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8652
Number of communities: 23
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8546
Number of communities: 26
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8448
Number of communities: 27
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 12217
Number of edges: 420063

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8272
Number of communities: 30
Elapsed time: 1 seconds</code></pre>
</div>
<div id="nscls-ccl19-frcs-and-nscls-tils" class="section level3">
<h3>NSCLS CCL19⁺ FRCs and NSCLS TILs</h3>
<pre class="r"><code>#Define color palet
palet &lt;- c(&quot;#1B9E77&quot;, &quot;#54B0E4&quot;,&quot;#E3BE00&quot;, &quot;#E41A1C&quot;, &quot;#4DAF4A&quot;,&quot;#FB9A99&quot;,&quot;#377EB8&quot;,&quot;#A65628&quot;,&quot;#222F75&quot;)
names(palet) &lt;- c( &quot;CAF2/TRC&quot;,&quot;CAF1/PRC&quot;,&quot;AdvFB&quot; ,&quot;SMC/PC&quot;, paste0(&quot;CD4&quot;, &quot;\u207A &quot;, &quot;T cells&quot;), paste0(&quot;CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;), &quot;B cells&quot;, &quot;Regulatory T cells&quot;,paste0(&quot;Cycling CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;))

DimPlot(merged_data, reduction = &quot;umap&quot;, group.by = &quot;cell_type&quot;, cols=palet)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank(),
         panel.grid.major = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;) + ggtitle(paste0(&quot;NSCLC CCL19&quot;, &quot;\U207A &quot;, &quot;FRCs&quot;, &quot; and TILs&quot;))</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/umap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-umap-1">
Past versions of umap-1.png
</button>
</p>
<div id="fig-umap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/umap-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-merged-data" class="section level3">
<h3>Save merged data</h3>
<pre class="r"><code>#saveRDS(merged_data, paste0(basedir,&quot;/data/Human/NSCLC_TILs_SI4.rds&quot;))</code></pre>
</div>
<div id="interactome-analysis-with-cellchat-suoqin-jin-et-al.-2021"
class="section level3">
<h3><strong>Interactome Analysis with Cellchat (Suoqin Jin et al.,
2021)</strong></h3>
</div>
<div id="convert-seurat-object-to-cellchat-object"
class="section level3">
<h3>Convert seurat object to cellchat object</h3>
<pre class="r"><code>cellchat &lt;- Cellchat_Analysis(merged_data)</code></pre>
<pre><code>[1] &quot;Create a CellChat object from a data matrix&quot;
Set cell identities for the new CellChat object 
The cell groups used for CellChat analysis are  AdvFB B cells CAF1/PRC CAF2/TRC CD4⁺ T cells CD8⁺ T cells Cycling CD8⁺ T cells Regulatory T cells SMC/PC </code></pre>
<pre class="r"><code>cellchat &lt;-CellChatDownstreamAnalysis(cellchat,&quot;human&quot;,thresh = 0.05)</code></pre>
<pre><code>triMean is used for calculating the average gene expression per cell group. 
[1] &quot;&gt;&gt;&gt; Run CellChat on sc/snRNA-seq data &lt;&lt;&lt; [2024-11-05 21:52:19.838019]&quot;
[1] &quot;&gt;&gt;&gt; CellChat inference is done. Parameter values are stored in `object@options$parameter` &lt;&lt;&lt; [2024-11-05 21:53:53.902528]&quot;</code></pre>
</div>
<div id="save-cellchat-data" class="section level3">
<h3>Save cellchat data</h3>
<pre class="r"><code>#saveRDS(cellchat,paste0(basedir,&quot;/data/Human/CCL19_FRC_TIL_SIF5_cellchat.rds&quot;))</code></pre>
</div>
<div id="interactome-analysis-supplementary-figure-3d"
class="section level3">
<h3>Interactome analysis (Supplementary Figure 3D)</h3>
<pre class="r"><code>gg &lt;- netAnalysis_signalingRole_scatter(cellchat,color.use = palet)
gg &lt;- gg + ggtitle(&quot;Interactome analysis (Cellchat)&quot;)
gg</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/interactome%20analysis-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-interactome-analysis-1">
Past versions of "interactome analysis-1.png"
</button>
</p>
<div id="fig-no-spaces-interactome-analysis-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/interactome analysis-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="outgoing-signaling-patterns" class="section level3">
<h3>Outgoing signaling patterns</h3>
<pre class="r"><code>selectK(cellchat, pattern = &quot;outgoing&quot;)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/outgoing%20signaling-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-outgoing-signaling-1">
Past versions of "outgoing signaling-1.png"
</button>
</p>
<div id="fig-no-spaces-outgoing-signaling-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/outgoing signaling-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="incoming-signaling-patterns" class="section level3">
<h3>Incoming signaling patterns</h3>
<pre class="r"><code>selectK(cellchat, pattern = &quot;incoming&quot;)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/incoming%20signaling-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-incoming-signaling-1">
Past versions of "incoming signaling-1.png"
</button>
</p>
<div id="fig-no-spaces-incoming-signaling-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/incoming signaling-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="incoming-and-outgoing-patterns" class="section level3">
<h3>Incoming and outgoing patterns</h3>
<pre class="r"><code>nPatterns &lt;- 7
cellchat &lt;- identifyCommunicationPatterns(cellchat, pattern = &quot;outgoing&quot;, k = nPatterns, color.use = palet)
cellchat &lt;- identifyCommunicationPatterns(cellchat, pattern = &quot;incoming&quot;, k = nPatterns, color.use = palet)</code></pre>
</div>
<div
id="joint-dotplot-for-incoming-and-outcoming-communication-patterns-supplementary-figure-3e"
class="section level3">
<h3>Joint dotplot for incoming and outcoming communication patterns
(Supplementary Figure 3E)</h3>
<pre class="r"><code>pathways &lt;-c(&quot;CXCL&quot;,&quot;CCL&quot;,&quot;ICAM&quot;,&quot;VCAM&quot;,&quot;MK&quot;,&quot;FGF&quot;,&quot;NECTIN&quot;,&quot;TIGIT&quot;,&quot;SELL&quot;,&quot;ITGB2&quot;,&quot;NOTCH&quot;,&quot;BAFF&quot;,&quot;SEMA3&quot;,&quot;SEMA4&quot;,&quot;LIGHT&quot;,&quot;EGF&quot;,&quot;IFN-II&quot;,&quot;CD226&quot;,&quot;IGF&quot;,&quot;FASLG&quot;)

order_list &lt;-c(&quot;CAF2/TRC&quot;,&quot;CAF1/PRC&quot;,&quot;SMC/PC&quot;,&quot;AdvFB&quot;,&quot;Regulatory T cells&quot;, &quot;CD4⁺ T cells&quot;,&quot;CD8⁺ T cells&quot;,&quot;Cycling CD8⁺ T cells&quot;,&quot;B cells&quot;)

netAnalysis_joint_dot(cellchat,color.use = palet,font.size = 12,pathways = pathways, order_list = order_list)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/joint%20dotplot%20in%20out%20signaling-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-joint-dotplot-in-out-signaling-1">
Past versions of "joint dotplot in out signaling-1.png"
</button>
</p>
<div id="fig-no-spaces-joint-dotplot-in-out-signaling-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/joint dotplot in out signaling-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="merge-ccl19-frcs-and-cd8-t-cell-subsets-in-nsclc"
class="section level2">
<h2><strong>Merge CCL19⁺ FRCs and CD8⁺ T cell subsets in
NSCLC</strong></h2>
<div id="read-cd8-t-cell-data" class="section level3">
<h3>Read CD8⁺ T cell data</h3>
<pre class="r"><code>NCLS_TIL_FRC_cd8 &lt;-readRDS(paste0(basedir,&quot;/data/Human/NSCLC_TILs_CD8_pop.rds&quot;))</code></pre>
</div>
<div id="merge-nscls-ccl19-frcs-and-nscls-cd8-t-cell-subsets"
class="section level3">
<h3>Merge NSCLS CCL19⁺ FRCs and NSCLS CD8⁺ T cell subsets</h3>
<pre class="r"><code>same_columns &lt;- intersect(colnames(NCLS_TIL_FRC_cd8@meta.data),colnames(NSCLC_CCL19_FRCs@meta.data))

NCLS_TIL_FRC_cd8@meta.data &lt;-NCLS_TIL_FRC_cd8@meta.data[,same_columns]
NSCLC_CCL19_FRCs@meta.data &lt;-NSCLC_CCL19_FRCs@meta.data[,same_columns]

merged_data&lt;- merge(NCLS_TIL_FRC_cd8, y = c(NSCLC_CCL19_FRCs),
             add.cell.ids = c(&#39;NCLS_TIL_FRC_cd8&#39;,&#39;NSCLC_CCL19_FRCs&#39;),
             project = &quot;NSCLC_FRC_TIL_CD8&quot;)

resolution &lt;- c(0.1, 0.25, 0.4, 0.6, 0.8, 1.,1.2,1.4,1.6,2.)
merged_data  &lt;- preprocessing(merged_data,resolution)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9616
Number of communities: 5
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9304
Number of communities: 9
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9108
Number of communities: 12
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8875
Number of communities: 16
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8721
Number of communities: 20
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8585
Number of communities: 22
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8473
Number of communities: 24
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8361
Number of communities: 23
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8245
Number of communities: 24
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10459
Number of edges: 356757

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8046
Number of communities: 30
Elapsed time: 0 seconds</code></pre>
</div>
<div id="nscls-ccl19-frcs-and-nscls-cd8-t-cell-subsets"
class="section level3">
<h3>NSCLS CCL19⁺ FRCs and NSCLS CD8⁺ T cell subsets</h3>
<pre class="r"><code>#Extend palet for CD8 T cell subsets
palet &lt;- c(&quot;#1B9E77&quot;, &quot;#54B0E4&quot;,&quot;#E3BE00&quot;, &quot;#E41A1C&quot;,&quot;#F8766D&quot;,&quot;#00C08B&quot;,&quot;#7CAE00&quot;,&quot;#00B4F0&quot;,&quot;#F564E3&quot;)
names(palet) &lt;- c(&quot;CAF2/TRC&quot;,&quot;CAF1/PRC&quot;,&quot;AdvFB&quot; ,&quot;SMC/PC&quot;,paste0(&quot;Stem-like/Naive CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),paste0(&quot;Exhausted CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;), paste0(&quot;Effector-memory CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),paste0(&quot;Effector CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),paste0(&quot;Cycling CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;))

palet &lt;- palet[names(palet) %in% unique(merged_data$cell_type)]

DimPlot(merged_data, reduction = &quot;umap&quot;, group.by = &quot;cell_type&quot;, cols = palet)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/umapNSCLS%20CCL19%20FRC%20CD8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umapNSCLS-CCL19-FRC-CD8-1">
Past versions of "umapNSCLS CCL19 FRC CD8-1.png"
</button>
</p>
<div id="fig-no-spaces-umapNSCLS-CCL19-FRC-CD8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/umapNSCLS CCL19 FRC CD8-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nscls-ccl19-frcs-and-nscls-cd8-t-cell-data"
class="section level3">
<h3>NSCLS CCL19⁺ FRCs and NSCLS CD8⁺ T cell data</h3>
<pre class="r"><code>#saveRDS(merged_data, paste0(basedir,&quot;/data/Human/NSCLC_TILs_CD8_pop_merge.rds&quot;))</code></pre>
</div>
<div id="interactome-analysis-with-cellchat-suoqin-jin-et-al.-2021-1"
class="section level3">
<h3><strong>Interactome Analysis with Cellchat (Suoqin Jin et al.,
2021)</strong></h3>
</div>
<div id="convert-seurat-object-to-cellchat-object-1"
class="section level3">
<h3>Convert seurat object to cellchat object</h3>
<pre class="r"><code>cellchat &lt;- Cellchat_Analysis(merged_data)</code></pre>
<pre><code>[1] &quot;Create a CellChat object from a data matrix&quot;
Set cell identities for the new CellChat object 
The cell groups used for CellChat analysis are  AdvFB CAF1/PRC CAF2/TRC Cycling CD8⁺ T cells Effector CD8⁺ T cells Effector-memory CD8⁺ T cells Exhausted CD8⁺ T cells SMC/PC Stem-like/Naive CD8⁺ T cells </code></pre>
<pre class="r"><code>cellchat &lt;-CellChatDownstreamAnalysis(cellchat,&quot;human&quot;,thresh=0.08)</code></pre>
<pre><code>triMean is used for calculating the average gene expression per cell group. 
[1] &quot;&gt;&gt;&gt; Run CellChat on sc/snRNA-seq data &lt;&lt;&lt; [2024-11-05 22:00:33.638104]&quot;
[1] &quot;&gt;&gt;&gt; CellChat inference is done. Parameter values are stored in `object@options$parameter` &lt;&lt;&lt; [2024-11-05 22:02:02.135343]&quot;</code></pre>
</div>
<div id="save-cellchat-data-1" class="section level3">
<h3>Save cellchat data</h3>
<pre class="r"><code>#saveRDS(cellchat,paste0(basedir,&quot;/data/Human/CCL19_FRC_TIL_F3E_cellchat.rds&quot;))</code></pre>
</div>
<div id="interactome-analysis-figure-3e" class="section level3">
<h3>Interactome analysis (Figure 3E)</h3>
<pre class="r"><code>gg &lt;- netAnalysis_signalingRole_scatter(cellchat,color.use = palet)
gg &lt;- gg + ggtitle(&quot;Interactome analysis (Cellchat)&quot;)
gg</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/interactome%20analysis%20NSCLC%20FRC%20CD8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-interactome-analysis-NSCLC-FRC-CD8-1">
Past versions of "interactome analysis NSCLC FRC CD8-1.png"
</button>
</p>
<div id="fig-no-spaces-interactome-analysis-NSCLC-FRC-CD8-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/interactome analysis NSCLC FRC CD8-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="outgoing-signaling-patterns-1" class="section level3">
<h3>Outgoing signaling patterns</h3>
<pre class="r"><code>selectK(cellchat, pattern = &quot;outgoing&quot;)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/outgoing%20signaling%20FRC%20CD8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-outgoing-signaling-FRC-CD8-1">
Past versions of "outgoing signaling FRC CD8-1.png"
</button>
</p>
<div id="fig-no-spaces-outgoing-signaling-FRC-CD8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/outgoing signaling FRC CD8-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="incoming-signaling-patterns-1" class="section level3">
<h3>Incoming signaling patterns</h3>
<pre class="r"><code>selectK(cellchat, pattern = &quot;incoming&quot;)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/incoming%20signaling%20FRC%20CD8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-incoming-signaling-FRC-CD8-1">
Past versions of "incoming signaling FRC CD8-1.png"
</button>
</p>
<div id="fig-no-spaces-incoming-signaling-FRC-CD8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/incoming signaling FRC CD8-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="incoming-and-outgoing-patterns-1" class="section level3">
<h3>Incoming and outgoing patterns</h3>
<pre class="r"><code>nPatterns &lt;- 8
cellchat &lt;- identifyCommunicationPatterns(cellchat, pattern = &quot;outgoing&quot;, k = nPatterns, color.use = palet)
cellchat &lt;- identifyCommunicationPatterns(cellchat, pattern = &quot;incoming&quot;, k = nPatterns, color.use = palet)</code></pre>
</div>
<div
id="joint-dotplot-for-incoming-and-outcoming-communication-patterns-figure-3f"
class="section level3">
<h3>Joint dotplot for incoming and outcoming communication patterns
(Figure 3F)</h3>
<pre class="r"><code>pathways &lt;-c(&quot;MHC-II&quot;,&quot;LAMININ&quot;,&quot;FN1&quot;,&quot;CXCL&quot;,&quot;CCL&quot;,&quot;VCAM&quot;,&quot;ICAM&quot;,&quot;ITGB2&quot;,&quot;LIGHT&quot;,&quot;NECTIN&quot;,&quot;TIGIT&quot;,&quot;CD226&quot;,&quot;EGF&quot;,&quot;FASLG&quot;,&quot;SEMA4&quot;,&quot;IFN-II&quot;,&quot;IL16&quot;,&quot;GDF&quot;)

order_list &lt;-c(&quot;CAF2/TRC&quot;,&quot;CAF1/PRC&quot;,paste0(&quot;Stem-like/Naive CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Exhausted CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Effector-memory CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Effector CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Cycling CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;))

netAnalysis_joint_dot(cellchat,color.use = palet,font.size = 8,pathways = pathways, order_list = order_list, exclude = c(&quot;AdvFB&quot; ,&quot;SMC/PC&quot;))</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/joint%20dotplot%20in%20out%20signaling%20FRC%20CD8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-joint-dotplot-in-out-signaling-FRC-CD8-1">
Past versions of "joint dotplot in out signaling FRC CD8-1.png"
</button>
</p>
<div id="fig-no-spaces-joint-dotplot-in-out-signaling-FRC-CD8-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/joint dotplot in out signaling FRC CD8-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="attraction-and-adhesion-figure-3g"
class="section level3 tabset">
<h3 class="tabset">Attraction and adhesion (Figure 3G)</h3>
<div id="heatmap-stroma" class="section level4">
<h4>Heatmap stroma</h4>
<pre class="r"><code>Selgenes &lt;- c(&quot;VCAM1&quot;,&quot;CXCL16&quot;, &quot;CD34&quot;,&quot;CXCL12&quot;)
order_list &lt;-c(&quot;CAF1/PRC&quot;,&quot;SMC/PC&quot;, &quot;CAF2/TRC&quot;,&quot;AdvFB&quot;)
data_subset &lt;- subset(merged_data, cell_type  %in% order_list)
  
Heatmap(data_subset, Selgenes, order_list = order_list, palet = palet)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/heatmap%20VCAM1%20CXCL16%20CD34%20CXCL12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-VCAM1-CXCL16-CD34-CXCL12-1">
Past versions of "heatmap VCAM1 CXCL16 CD34 CXCL12-1.png"
</button>
</p>
<div id="fig-no-spaces-heatmap-VCAM1-CXCL16-CD34-CXCL12-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/heatmap VCAM1 CXCL16 CD34 CXCL12-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heatmap-cd8-t-cell" class="section level4">
<h4>Heatmap CD8⁺ T cell</h4>
<pre class="r"><code>Selgenes &lt;- c(&quot;ITGA4&quot;,&quot;ITGB1&quot;,&quot;ITGB7&quot;,&quot;CXCR6&quot;,&quot;SELL&quot;,&quot;CXCR4&quot;)
order_list &lt;-c(paste0(&quot;Stem-like/Naive CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;), 
               paste0(&quot;Effector-memory CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Effector CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Exhausted CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Cycling CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;) )
data_subset &lt;- subset(merged_data, cell_type  %in% order_list)

Heatmap(data_subset, Selgenes, order_list = order_list, palet = palet)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/heatmap%20ITGA4%20ITGB1%20ITGB7%20SELL%20CXCR4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-ITGA4-ITGB1-ITGB7-SELL-CXCR4-1">
Past versions of "heatmap ITGA4 ITGB1 ITGB7 SELL CXCR4-1.png"
</button>
</p>
<div id="fig-no-spaces-heatmap-ITGA4-ITGB1-ITGB7-SELL-CXCR4-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/heatmap ITGA4 ITGB1 ITGB7 SELL CXCR4-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="t-cell-reactivity-figure-3h" class="section level3 tabset">
<h3 class="tabset">T cell reactivity (Figure 3H)</h3>
<div id="heatmap-stroma-1" class="section level4">
<h4>Heatmap stroma</h4>
<pre class="r"><code>Selgenes &lt;- c(&quot;NECTIN3&quot;,&quot;NECTIN2&quot;,&quot;EGFR&quot;,&quot;FAS&quot;,&quot;IFNGR1&quot;,&quot;IFNGR2&quot;)
order_list &lt;-c(&quot;CAF1/PRC&quot;,&quot;SMC/PC&quot;, &quot;CAF2/TRC&quot;,&quot;AdvFB&quot;)
data_subset &lt;- subset(merged_data, cell_type  %in% order_list)
  
Heatmap(data_subset, Selgenes, order_list = order_list, palet = palet)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/heatmap%20NECTIN3%20NECTIN2%20EGFR%20FAS%20IFNGR1%20IFNGR2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-NECTIN3-NECTIN2-EGFR-FAS-IFNGR1-IFNGR2-1">
Past versions of "heatmap NECTIN3 NECTIN2 EGFR FAS IFNGR1 IFNGR2-1.png"
</button>
</p>
<div id="fig-no-spaces-heatmap-NECTIN3-NECTIN2-EGFR-FAS-IFNGR1-IFNGR2-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/heatmap NECTIN3 NECTIN2 EGFR FAS IFNGR1 IFNGR2-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heatmap-cd8-t-cell-1" class="section level4">
<h4>Heatmap CD8⁺ T cell</h4>
<pre class="r"><code>Selgenes &lt;- c(&quot;TIGIT&quot;,&quot;CD226&quot;,&quot;AREG&quot;,&quot;FASLG&quot;,&quot;IFNG&quot;)
order_list &lt;-c(paste0(&quot;Stem-like/Naive CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;), 
               paste0(&quot;Effector-memory CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Effector CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Exhausted CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Cycling CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;) )
data_subset &lt;- subset(merged_data, cell_type  %in% order_list)

Heatmap(data_subset, Selgenes, order_list = order_list, palet = palet)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/heatmap%20TIGIT%20CD226%20AREG%20FASLG%20IFNG-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-TIGIT-CD226-AREG-FASLG-IFNG-1">
Past versions of "heatmap TIGIT CD226 AREG FASLG IFNG-1.png"
</button>
</p>
<div id="fig-no-spaces-heatmap-TIGIT-CD226-AREG-FASLG-IFNG-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/heatmap TIGIT CD226 AREG FASLG IFNG-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="niche-formation-figure-3i" class="section level3 tabset">
<h3 class="tabset">Niche formation (Figure 3I)</h3>
<div id="heatmap-stroma-2" class="section level4">
<h4>Heatmap stroma</h4>
<pre class="r"><code>Selgenes &lt;- c(&quot;LTBR&quot;,&quot;TNFRSF14&quot;,&quot;NOTCH3&quot;,&quot;JAG1&quot;)
order_list &lt;-c(&quot;CAF1/PRC&quot;,&quot;SMC/PC&quot;, &quot;CAF2/TRC&quot;,&quot;AdvFB&quot;)
data_subset &lt;- subset(merged_data, cell_type  %in% order_list)
  
Heatmap(data_subset, Selgenes, order_list = order_list, palet = palet)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/heatmap%20LTBR%20TNFRSF14%20NOTCH3%20JAG1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-LTBR-TNFRSF14-NOTCH3-JAG1-1">
Past versions of "heatmap LTBR TNFRSF14 NOTCH3 JAG1-1.png"
</button>
</p>
<div id="fig-no-spaces-heatmap-LTBR-TNFRSF14-NOTCH3-JAG1-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/heatmap LTBR TNFRSF14 NOTCH3 JAG1-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="heatmap-cd8-t-cell-2" class="section level4">
<h4>Heatmap CD8⁺ T cell</h4>
<pre class="r"><code>Selgenes &lt;- c(&quot;LTA&quot;,&quot;LTB&quot;,&quot;TNFSF14&quot;,&quot;DLL1&quot;,&quot;JAG1&quot;)
order_list &lt;-c(paste0(&quot;Stem-like/Naive CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;), 
               paste0(&quot;Effector-memory CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Effector CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Exhausted CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;),
               paste0(&quot;Cycling CD8&quot;, &quot;\u207A &quot;, &quot;T cells&quot;) )
data_subset &lt;- subset(merged_data, cell_type  %in% order_list)

Heatmap(data_subset, Selgenes, order_list = order_list, palet = palet)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/heatmap%20LTA%20LTB%20TNFSF14%20DLL1%20JAG1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-LTA-LTB-TNFSF14-DLL1-JAG1-1">
Past versions of "heatmap LTA LTB TNFSF14 DLL1 JAG1-1.png"
</button>
</p>
<div id="fig-no-spaces-heatmap-LTA-LTB-TNFSF14-DLL1-JAG1-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/heatmap LTA LTB TNFSF14 DLL1 JAG1-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="heatmaps-figure-3j" class="section level3 tabset">
<h3 class="tabset">Heatmaps (Figure 3J)</h3>
<div id="ltbr-regulated" class="section level4">
<h4>LTBR regulated</h4>
<pre class="r"><code>Selgenes &lt;- c(&quot;CLU&quot;,&quot;TNFSF13B&quot;,&quot;MFGE8&quot;)
order_list &lt;-c(&quot;CAF1/PRC&quot;,&quot;SMC/PC&quot;, &quot;CAF2/TRC&quot;,&quot;AdvFB&quot;)
data_subset &lt;- subset(merged_data, cell_type  %in% order_list)
  
Heatmap(data_subset, Selgenes, order_list = order_list, palet = palet)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/heatmap%20CLU%20TNFSF13B%20MFGE8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-CLU-TNFSF13B-MFGE8-1">
Past versions of "heatmap CLU TNFSF13B MFGE8-1.png"
</button>
</p>
<div id="fig-no-spaces-heatmap-CLU-TNFSF13B-MFGE8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/heatmap CLU TNFSF13B MFGE8-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tnfrsf14-regulated" class="section level4">
<h4>TNFRSF14 regulated</h4>
<pre class="r"><code>Selgenes &lt;- c(&quot;DKK1&quot;, &quot;IL32&quot;, &quot;IL34&quot;)
order_list &lt;-c(&quot;CAF1/PRC&quot;,&quot;SMC/PC&quot;, &quot;CAF2/TRC&quot;,&quot;AdvFB&quot;)
data_subset &lt;- subset(merged_data, cell_type  %in% order_list)
  
Heatmap(data_subset, Selgenes, order_list = order_list, palet = palet)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/heatmap%20DKK1%20IL32%20IL34-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-DKK1-IL32-IL34-1">
Past versions of "heatmap DKK1 IL32 IL34-1.png"
</button>
</p>
<div id="fig-no-spaces-heatmap-DKK1-IL32-IL34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/heatmap DKK1 IL32 IL34-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="notch3-regulated" class="section level4">
<h4>NOTCH3 regulated</h4>
<pre class="r"><code>Selgenes &lt;- c(&quot;HEY1&quot;, &quot;HEY2&quot;, &quot;HEYL&quot;)
order_list &lt;-c(&quot;CAF1/PRC&quot;,&quot;SMC/PC&quot;, &quot;CAF2/TRC&quot;,&quot;AdvFB&quot;)
data_subset &lt;- subset(merged_data, cell_type  %in% order_list)
  
Heatmap(data_subset, Selgenes, order_list = order_list, palet = palet)</code></pre>
<p><img src="figure/merge_NSCLC_FRC_TIL.Rmd/heatmap%20HEY1%20HEY2%20HEYL-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-HEY1-HEY2-HEYL-1">
Past versions of "heatmap HEY1 HEY2 HEYL-1.png"
</button>
</p>
<div id="fig-no-spaces-heatmap-HEY1-HEY2-HEYL-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/merge_NSCLC_FRC_TIL.Rmd/heatmap HEY1 HEY2 HEYL-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="session-info" class="section level3">
<h3>Session info</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.6.9

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] doParallel_1.0.17   iterators_1.0.14    foreach_1.5.2      
 [4] pheatmap_1.0.12     NMF_0.26            cluster_2.1.4      
 [7] rngtools_1.5.2      registry_0.5-1      CellChat_1.6.1     
[10] Biobase_2.60.0      BiocGenerics_0.46.0 igraph_1.5.0.1     
[13] dplyr_1.1.2         gridExtra_2.3       dittoSeq_1.12.1    
[16] ggplot2_3.4.2       Matrix_1.6-0        SeuratObject_4.1.3 
[19] Seurat_4.3.0.1      patchwork_1.1.2     stringr_1.5.0      
[22] purrr_1.0.1         here_1.0.1          magrittr_2.0.3     
[25] circlize_0.4.15     tidyr_1.3.0         tibble_3.2.1       
[28] workflowr_1.7.1    

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21            splines_4.3.1              
  [3] later_1.3.1                 bitops_1.0-7               
  [5] polyclip_1.10-4             ggnetwork_0.5.12           
  [7] lifecycle_1.0.3             rstatix_0.7.2              
  [9] rprojroot_2.0.3             globals_0.16.2             
 [11] processx_3.8.2              lattice_0.21-8             
 [13] MASS_7.3-60                 backports_1.4.1            
 [15] plotly_4.10.2               sass_0.4.7                 
 [17] rmarkdown_2.23              jquerylib_0.1.4            
 [19] yaml_2.3.7                  httpuv_1.6.11              
 [21] sctransform_0.3.5           sp_2.0-0                   
 [23] spatstat.sparse_3.0-2       reticulate_1.36.1          
 [25] cowplot_1.1.1               pbapply_1.7-2              
 [27] RColorBrewer_1.1-3          abind_1.4-5                
 [29] zlibbioc_1.46.0             Rtsne_0.16                 
 [31] GenomicRanges_1.52.0        RCurl_1.98-1.12            
 [33] git2r_0.33.0                GenomeInfoDbData_1.2.10    
 [35] IRanges_2.34.1              S4Vectors_0.38.1           
 [37] ggrepel_0.9.3               irlba_2.3.5.1              
 [39] listenv_0.9.0               spatstat.utils_3.1-0       
 [41] RSpectra_0.16-1             goftest_1.2-3              
 [43] spatstat.random_3.1-5       fitdistrplus_1.1-11        
 [45] parallelly_1.36.0           svglite_2.1.1              
 [47] leiden_0.4.3                codetools_0.2-19           
 [49] DelayedArray_0.28.0         tidyselect_1.2.0           
 [51] shape_1.4.6                 farver_2.1.1               
 [53] matrixStats_1.0.0           stats4_4.3.1               
 [55] spatstat.explore_3.2-1      jsonlite_1.8.7             
 [57] GetoptLong_1.0.5            BiocNeighbors_1.18.0       
 [59] ellipsis_0.3.2              progressr_0.13.0           
 [61] ggridges_0.5.4              ggalluvial_0.12.5          
 [63] survival_3.5-5              systemfonts_1.0.4          
 [65] tools_4.3.1                 ragg_1.2.5                 
 [67] sna_2.7-1                   ica_1.0-3                  
 [69] Rcpp_1.0.11                 glue_1.6.2                 
 [71] SparseArray_1.2.4           xfun_0.39                  
 [73] MatrixGenerics_1.12.3       GenomeInfoDb_1.36.1        
 [75] withr_2.5.0                 BiocManager_1.30.21.1      
 [77] fastmap_1.1.1               fansi_1.0.4                
 [79] callr_3.7.3                 digest_0.6.33              
 [81] R6_2.5.1                    mime_0.12                  
 [83] textshaping_0.3.6           colorspace_2.1-0           
 [85] Cairo_1.6-1                 scattermore_1.2            
 [87] tensor_1.5                  spatstat.data_3.0-1        
 [89] utf8_1.2.3                  generics_0.1.3             
 [91] data.table_1.14.8           FNN_1.1.3.2                
 [93] httr_1.4.6                  htmlwidgets_1.6.2          
 [95] S4Arrays_1.2.1              whisker_0.4.1              
 [97] uwot_0.1.16                 pkgconfig_2.0.3            
 [99] gtable_0.3.3                ComplexHeatmap_2.16.0      
[101] lmtest_0.9-40               SingleCellExperiment_1.22.0
[103] XVector_0.40.0              htmltools_0.5.5            
[105] carData_3.0-5               clue_0.3-64                
[107] scales_1.2.1                png_0.1-8                  
[109] knitr_1.43                  rstudioapi_0.15.0          
[111] rjson_0.2.21                reshape2_1.4.4             
[113] coda_0.19-4                 statnet.common_4.9.0       
[115] nlme_3.1-162                cachem_1.0.8               
[117] zoo_1.8-12                  GlobalOptions_0.1.2        
[119] KernSmooth_2.23-22          miniUI_0.1.1.1             
[121] pillar_1.9.0                grid_4.3.1                 
[123] vctrs_0.6.3                 RANN_2.6.1                 
[125] ggpubr_0.6.0                promises_1.2.0.1           
[127] car_3.1-2                   xtable_1.8-4               
[129] evaluate_0.21               cli_3.6.1                  
[131] compiler_4.3.1              rlang_1.1.1                
[133] crayon_1.5.2                ggsignif_0.6.4             
[135] future.apply_1.11.0         labeling_0.4.2             
[137] ps_1.7.5                    forcats_1.0.0              
[139] getPass_0.2-4               plyr_1.8.8                 
[141] fs_1.6.3                    stringi_1.7.12             
[143] network_1.18.1              BiocParallel_1.34.2        
[145] viridisLite_0.4.2           deldir_1.0-9               
[147] gridBase_0.4-7              munsell_0.5.0              
[149] lazyeval_0.2.2              spatstat.geom_3.2-4        
[151] future_1.33.0               shiny_1.7.4.1              
[153] highr_0.10                  SummarizedExperiment_1.30.2
[155] ROCR_1.0-11                 broom_1.0.5                
[157] bslib_0.5.0                </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Tue Nov  5 22:05:11 2024&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.6.9

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] doParallel_1.0.17   iterators_1.0.14    foreach_1.5.2      
 [4] pheatmap_1.0.12     NMF_0.26            cluster_2.1.4      
 [7] rngtools_1.5.2      registry_0.5-1      CellChat_1.6.1     
[10] Biobase_2.60.0      BiocGenerics_0.46.0 igraph_1.5.0.1     
[13] dplyr_1.1.2         gridExtra_2.3       dittoSeq_1.12.1    
[16] ggplot2_3.4.2       Matrix_1.6-0        SeuratObject_4.1.3 
[19] Seurat_4.3.0.1      patchwork_1.1.2     stringr_1.5.0      
[22] purrr_1.0.1         here_1.0.1          magrittr_2.0.3     
[25] circlize_0.4.15     tidyr_1.3.0         tibble_3.2.1       
[28] workflowr_1.7.1    

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21            splines_4.3.1              
  [3] later_1.3.1                 bitops_1.0-7               
  [5] polyclip_1.10-4             ggnetwork_0.5.12           
  [7] lifecycle_1.0.3             rstatix_0.7.2              
  [9] rprojroot_2.0.3             globals_0.16.2             
 [11] processx_3.8.2              lattice_0.21-8             
 [13] MASS_7.3-60                 backports_1.4.1            
 [15] plotly_4.10.2               sass_0.4.7                 
 [17] rmarkdown_2.23              jquerylib_0.1.4            
 [19] yaml_2.3.7                  httpuv_1.6.11              
 [21] sctransform_0.3.5           sp_2.0-0                   
 [23] spatstat.sparse_3.0-2       reticulate_1.36.1          
 [25] cowplot_1.1.1               pbapply_1.7-2              
 [27] RColorBrewer_1.1-3          abind_1.4-5                
 [29] zlibbioc_1.46.0             Rtsne_0.16                 
 [31] GenomicRanges_1.52.0        RCurl_1.98-1.12            
 [33] git2r_0.33.0                GenomeInfoDbData_1.2.10    
 [35] IRanges_2.34.1              S4Vectors_0.38.1           
 [37] ggrepel_0.9.3               irlba_2.3.5.1              
 [39] listenv_0.9.0               spatstat.utils_3.1-0       
 [41] RSpectra_0.16-1             goftest_1.2-3              
 [43] spatstat.random_3.1-5       fitdistrplus_1.1-11        
 [45] parallelly_1.36.0           svglite_2.1.1              
 [47] leiden_0.4.3                codetools_0.2-19           
 [49] DelayedArray_0.28.0         tidyselect_1.2.0           
 [51] shape_1.4.6                 farver_2.1.1               
 [53] matrixStats_1.0.0           stats4_4.3.1               
 [55] spatstat.explore_3.2-1      jsonlite_1.8.7             
 [57] GetoptLong_1.0.5            BiocNeighbors_1.18.0       
 [59] ellipsis_0.3.2              progressr_0.13.0           
 [61] ggridges_0.5.4              ggalluvial_0.12.5          
 [63] survival_3.5-5              systemfonts_1.0.4          
 [65] tools_4.3.1                 ragg_1.2.5                 
 [67] sna_2.7-1                   ica_1.0-3                  
 [69] Rcpp_1.0.11                 glue_1.6.2                 
 [71] SparseArray_1.2.4           xfun_0.39                  
 [73] MatrixGenerics_1.12.3       GenomeInfoDb_1.36.1        
 [75] withr_2.5.0                 BiocManager_1.30.21.1      
 [77] fastmap_1.1.1               fansi_1.0.4                
 [79] callr_3.7.3                 digest_0.6.33              
 [81] R6_2.5.1                    mime_0.12                  
 [83] textshaping_0.3.6           colorspace_2.1-0           
 [85] Cairo_1.6-1                 scattermore_1.2            
 [87] tensor_1.5                  spatstat.data_3.0-1        
 [89] utf8_1.2.3                  generics_0.1.3             
 [91] data.table_1.14.8           FNN_1.1.3.2                
 [93] httr_1.4.6                  htmlwidgets_1.6.2          
 [95] S4Arrays_1.2.1              whisker_0.4.1              
 [97] uwot_0.1.16                 pkgconfig_2.0.3            
 [99] gtable_0.3.3                ComplexHeatmap_2.16.0      
[101] lmtest_0.9-40               SingleCellExperiment_1.22.0
[103] XVector_0.40.0              htmltools_0.5.5            
[105] carData_3.0-5               clue_0.3-64                
[107] scales_1.2.1                png_0.1-8                  
[109] knitr_1.43                  rstudioapi_0.15.0          
[111] rjson_0.2.21                reshape2_1.4.4             
[113] coda_0.19-4                 statnet.common_4.9.0       
[115] nlme_3.1-162                cachem_1.0.8               
[117] zoo_1.8-12                  GlobalOptions_0.1.2        
[119] KernSmooth_2.23-22          miniUI_0.1.1.1             
[121] pillar_1.9.0                grid_4.3.1                 
[123] vctrs_0.6.3                 RANN_2.6.1                 
[125] ggpubr_0.6.0                promises_1.2.0.1           
[127] car_3.1-2                   xtable_1.8-4               
[129] evaluate_0.21               cli_3.6.1                  
[131] compiler_4.3.1              rlang_1.1.1                
[133] crayon_1.5.2                ggsignif_0.6.4             
[135] future.apply_1.11.0         labeling_0.4.2             
[137] ps_1.7.5                    forcats_1.0.0              
[139] getPass_0.2-4               plyr_1.8.8                 
[141] fs_1.6.3                    stringi_1.7.12             
[143] network_1.18.1              BiocParallel_1.34.2        
[145] viridisLite_0.4.2           deldir_1.0-9               
[147] gridBase_0.4-7              munsell_0.5.0              
[149] lazyeval_0.2.2              spatstat.geom_3.2-4        
[151] future_1.33.0               shiny_1.7.4.1              
[153] highr_0.10                  SummarizedExperiment_1.30.2
[155] ROCR_1.0-11                 broom_1.0.5                
[157] bslib_0.5.0                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
