<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Chrysa Papadopoulou" />


<title>CCL19-expressing FRC subsets in coronavirus-vector based immunotherapy</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CCL19_FRCs_lung_cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CCL19-expressing FRC subsets in
coronavirus-vector based immunotherapy</h1>
<h4 class="author">Chrysa Papadopoulou</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CCL19_FRCs_lung_cancer/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240808code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240808)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240808code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240808)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomPchryssaCCL19FRCslungcancertree41217c2c204930f1e6849551e73897937761ca3atargetblank41217c2a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/tree/41217c2c204930f1e6849551e73897937761ca3a" target="_blank">41217c2</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomPchryssaCCL19FRCslungcancertree41217c2c204930f1e6849551e73897937761ca3atargetblank41217c2a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/tree/41217c2c204930f1e6849551e73897937761ca3a" target="_blank">41217c2</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    data/Final_submission/
    Ignored:    data/Human/
    Ignored:    data/Mouse/
    Ignored:    data/Public/
    Ignored:    output/GSEA_AdvFB_SULF1/
    Ignored:    output/GSEA_AdvFB_TLS/
    Ignored:    output/GSEA_CCR7_T/
    Ignored:    output/GSEA_CD8_T/
    Ignored:    output/GSEA_CYCL_T/
    Ignored:    output/GSEA_EXH_T/
    Ignored:    output/GSEA_SMC_PRC/

Untracked files:
    Untracked:  README.html
    Untracked:  analysis/.h5seurat
    Untracked:  analysis/Compare_tumors.Rmd
    Untracked:  analysis/NSCLC_PDAC_CAFs.Rmd
    Untracked:  analysis/NSCLC_vs_BRCA_CAFs.Rmd
    Untracked:  analysis/Seurat_to_SCE.Rmd
    Untracked:  analysis/Total_CD8_DTR_Tetra.Rmd
    Untracked:  analysis/compression.Rmd
    Untracked:  analysis/index_hidden.Rmd
    Untracked:  analysis/mcov_R.Rmd

Unstaged changes:
    Modified:   analysis/NSCLC_TILS.Rmd
    Modified:   analysis/NSCLC_stroma_CCCL19_FRC.Rmd
    Modified:   analysis/NSCLC_trajectory_analysis.Rmd
    Modified:   analysis/NSCLC_vs_Tonsils.Rmd
    Modified:   analysis/Tetra_CD8_T_cells.Rmd
    Modified:   analysis/merge_NSCLC_FRC_TIL.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Mouse_mCOV.Rmd</code>) and HTML
(<code>docs/Mouse_mCOV.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/41217c2c204930f1e6849551e73897937761ca3a/analysis/Mouse_mCOV.Rmd" target="_blank">41217c2</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-09-23
</td>
<td>
Modify figure order
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Pchryssa/CCL19_FRCs_lung_cancer/31727316684c527736daf804c64e3544b47c5c47/docs/Mouse_mCOV.html" target="_blank">3172731</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/09ceaf16815aebf5fbe66f9ae5d95e1757b310ce/analysis/Mouse_mCOV.Rmd" target="_blank">09ceaf1</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-21
</td>
<td>
mCOV add thresh
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Pchryssa/CCL19_FRCs_lung_cancer/e8525456727c7d3309a773505aefe7601ecb2af3/docs/Mouse_mCOV.html" target="_blank">e852545</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/512f20daaa15b8e4489f38b086ce0e12d31b8cf6/analysis/Mouse_mCOV.Rmd" target="_blank">512f20d</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-21
</td>
<td>
Mouse mCOV
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>suppressPackageStartupMessages({
  library(here)
  library(purrr)
  library(dplyr)
  library(stringr)
  library(patchwork)
  library(Seurat)
  library(Matrix)
  library(dittoSeq)
  library(gridExtra)
  library(gsubfn)
  library(ggsci)
  library(slingshot)
  library(tradeSeq)
  library(mgcv)
})</code></pre>
</div>
<div id="set-directory" class="section level2">
<h2>Set directory</h2>
<pre class="r"><code>basedir &lt;- here()</code></pre>
<div id="read-ccl19-eyfp-mcov-fit31-g33-cell-data"
class="section level3">
<h3>Read CCL19-EYFP⁺ mCOV-FIt31-g33 cell data</h3>
<pre class="r"><code>CCL19_EYFP_mCOV &lt;- readRDS(paste0(basedir,&quot;/data/Mouse/mCOV.rds&quot;))</code></pre>
</div>
<div id="define-color-palette" class="section level3">
<h3>Define color palette</h3>
<pre class="r"><code>cols &lt;- c(&quot;#C77CFF&quot;,&quot;#F8766D&quot;,&quot;#00BA38&quot;,&quot;#B79F00&quot;,&quot;#FF64B0&quot;,&quot;#00BFC4&quot;,&quot;#00B4F0&quot;,&quot;#7CAE00&quot;)
names(cols) &lt;-c(paste0(&quot;Smoc1&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),paste0(&quot;Cd34&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),paste0(&quot;Npnt&quot;, expression(&quot;\u207A &quot;), &quot;AlvFB&quot;),paste0(&quot;Hhip&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),paste0(&quot;Sulf1&quot;, expression(&quot;\u207A &quot;), &quot;TRC&quot;),paste0(&quot;Rgs5&quot;, expression(&quot;\u207A &quot;), &quot;PRC&quot;),&quot;SMC/PC&quot;,&quot;TLS TRC&quot;)</code></pre>
</div>
<div id="ccl19-eyfp-mcov-fit31-g33-cells" class="section level3">
<h3>CCL19-EYFP⁺ mCOV-FIt31-g33 cells</h3>
<div id="umap-colored-per-celltype-supplementary-figure-9c"
class="section level4">
<h4>Umap colored per celltype (Supplementary Figure 9C)</h4>
<pre class="r"><code>DimPlot(CCL19_EYFP_mCOV, reduction = &quot;umap&quot;, group.by = &quot;annot&quot;, cols=cols)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;) + ggtitle(paste0(&quot;Ccl19-EYFP&quot;, &quot;\U207A &quot;, &quot;cells (mCOV-FIt31-g33 cells)&quot;))</code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/CCL19%20EYFP%20umap%20mCOV-FIt31-g33%20cells-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-CCL19-EYFP-umap-mCOV-FIt31-g33-cells-1">
Past versions of "CCL19 EYFP umap mCOV-FIt31-g33 cells-1.png"
</button>
</p>
<div id="fig-no-spaces-CCL19-EYFP-umap-mCOV-FIt31-g33-cells-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/CCL19 EYFP umap mCOV-FIt31-g33 cells-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="dotplot-figure-6f" class="section level3 tabset">
<h3 class="tabset">Dotplot (Figure 6F)</h3>
<pre class="r"><code>data_conv &lt;-CCL19_EYFP_mCOV
data_conv &lt;-Remove_ensebl_id(data_conv)

Idents(data_conv) &lt;- data_conv$annot
levels(data_conv)&lt;-levels(data_conv)[order(match(levels(data_conv),c(paste0(&quot;Cd34&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),paste0(&quot;Smoc1&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),paste0(&quot;Hhip&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),paste0(&quot;Npnt&quot;, expression(&quot;\u207A &quot;), &quot;AlvFB&quot;),paste0(&quot;Sulf1&quot;, expression(&quot;\u207A &quot;), &quot;TRC&quot;),paste0(&quot;Rgs5&quot;, expression(&quot;\u207A &quot;), &quot;PRC&quot;),&quot;SMC/PC&quot;,&quot;TLS TRC&quot;)))]
data_conv$annot &lt;- factor(as.character(data_conv@active.ident), levels = rev(c(paste0(&quot;Cd34&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),paste0(&quot;Smoc1&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;), paste0(&quot;Hhip&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;),paste0(&quot;Npnt&quot;, expression(&quot;\u207A &quot;), &quot;AlvFB&quot;),paste0(&quot;Sulf1&quot;, expression(&quot;\u207A &quot;), &quot;TRC&quot;),paste0(&quot;Rgs5&quot;, expression(&quot;\u207A &quot;), &quot;PRC&quot;),&quot;SMC/PC&quot;,&quot;TLS TRC&quot;)))

gene_list &lt;-c(&quot;Cd34&quot;,&quot;Pdpn&quot;,&quot;Hhip&quot;,&quot;Sulf1&quot;,&quot;Ccl11&quot;,&quot;Cxcl9&quot;,&quot;Tnc&quot;,&quot;Pdgfrb&quot;,&quot;Notch3&quot;,&quot;Rgs5&quot;,&quot;Acta2&quot;,&quot;Des&quot;,&quot;Myh11&quot;,&quot;Il33&quot;,&quot;Ccl19&quot;,&quot;Ccl21a&quot;,&quot;Cxcl13&quot;,&quot;Clu&quot;)

gg &lt;-DotPlot(data_conv, features = gene_list, group.by = &quot;annot&quot;, scale = TRUE, cols = c(&quot;lightgrey&quot;, &quot;#C51B7D&quot;), 
             scale.min = 0, scale.max = 100,col.min = 0, col.max = 8 , dot.scale = 4) + xlab(&quot; &quot;) + ylab(&quot; &quot;)

gg + theme(axis.text.x = element_text(angle = 45,hjust = 1))</code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/dotplot%20mouse%20tumor%20mCOV%20scale%20true-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-dotplot-mouse-tumor-mCOV-scale-true-1">
Past versions of "dotplot mouse tumor mCOV scale true-1.png"
</button>
</p>
<div id="fig-no-spaces-dotplot-mouse-tumor-mCOV-scale-true-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/dotplot mouse tumor mCOV scale true-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="differentiation-trajectories-on-ccl19-eyfp-mcov-fit31-g33-cells-based-on-slingshot-algorithm-figure-6g"
class="section level3 tabset">
<h3 class="tabset">Differentiation trajectories on CCL19-EYFP⁺
mCOV-FIt31-g33 cells based on Slingshot algorithm (Figure 6G)</h3>
<div
id="annotation-from-rna_snn_res.0.25-to-perform-trajectory-analysis"
class="section level4">
<h4>Annotation from RNA_snn_res.0.25 to perform trajectory analysis</h4>
<pre class="r"><code>#Set color palette
palet_new &lt;- c(&quot;#C77CFF&quot;,&quot;#F8766D&quot;,&quot;#00BA38&quot;,&quot;#B79F00&quot;,&quot;#FF64B0&quot;,&quot;#00BFC4&quot;,&quot;#00B4F0&quot;,&quot;#7CAE00&quot;)
names(palet_new) &lt;-c(2,1,4,3,5,6,7,8)

CCL19_EYFP_mCOV$annot_numeric &lt;- CCL19_EYFP_mCOV$annot
CCL19_EYFP_mCOV$annot_numeric[which(CCL19_EYFP_mCOV$annot == paste0(&quot;Cd34&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;))] &lt;- 1
CCL19_EYFP_mCOV$annot_numeric[which(CCL19_EYFP_mCOV$annot == paste0(&quot;Smoc1&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;))] &lt;- 2
CCL19_EYFP_mCOV$annot_numeric[which(CCL19_EYFP_mCOV$annot == paste0(&quot;Hhip&quot;, expression(&quot;\u207A &quot;), &quot;AdvFB&quot;))] &lt;- 3
CCL19_EYFP_mCOV$annot_numeric[which(CCL19_EYFP_mCOV$annot == paste0(&quot;Npnt&quot;, expression(&quot;\u207A &quot;), &quot;AlvFB&quot;))] &lt;- 4
CCL19_EYFP_mCOV$annot_numeric[which(CCL19_EYFP_mCOV$annot == paste0(&quot;Sulf1&quot;, expression(&quot;\u207A &quot;), &quot;TRC&quot;))] &lt;- 5
CCL19_EYFP_mCOV$annot_numeric[which(CCL19_EYFP_mCOV$annot == paste0(&quot;Rgs5&quot;, expression(&quot;\u207A &quot;), &quot;PRC&quot;))] &lt;- 6
CCL19_EYFP_mCOV$annot_numeric[which(CCL19_EYFP_mCOV$annot == &quot;SMC/PC&quot;)] &lt;- 7
CCL19_EYFP_mCOV$annot_numeric[which(CCL19_EYFP_mCOV$annot == &quot;TLS TRC&quot;)] &lt;- 8

DimPlot(CCL19_EYFP_mCOV, reduction = &quot;umap&quot;, group.by = &quot;annot_numeric&quot;, cols = palet_new)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(),
        panel.grid.minor = element_blank(),
        panel.grid.major = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;) + ggtitle(&quot;New annotation&quot;)</code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/umap%20CCL19%20EYFP%20umap%20map%20annotation%20%20mCOV-FIt31-g33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umap-CCL19-EYFP-umap-map-annotation--mCOV-FIt31-g33-1">
Past versions of "umap CCL19 EYFP umap map annotation
mCOV-FIt31-g33-1.png"
</button>
</p>
<div
id="fig-no-spaces-umap-CCL19-EYFP-umap-map-annotation--mCOV-FIt31-g33-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/umap CCL19 EYFP umap map annotation  mCOV-FIt31-g33-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div
id="differentiation-trajectories-for-sulf1-trc-tls-trc-and-rgs5-prc"
class="section level4">
<h4>Differentiation trajectories for Sulf1⁺ TRC, TLS TRC and Rgs5⁺
PRC</h4>
<pre class="r"><code>#Calculation of CCL19-EYFP`r knitr::asis_output(&quot;\U207A&quot;)` mCOV-FIt31-g33 cell differentiation trajectories
clustering &lt;- CCL19_EYFP_mCOV@meta.data$annot_numeric
dimred &lt;- CCL19_EYFP_mCOV@reductions$umap@cell.embeddings

# Slingshot for TRC (Sulf1, TLS)
pto_TRC &lt;- slingshot(dimred, clustering, start.clus = &#39;2&#39;, end.clus = &#39;8&#39;, reducedDim = &#39;umap&#39;,extend=&quot;n&quot;,stretch=0.07,thresh=0.05)
pto_TRC &lt;- as.SlingshotDataSet(pto_TRC)

# Slingshot for PRC
pto_PRC &lt;- slingshot(dimred, clustering, start.clus = &#39;7&#39;, end.clus = &#39;6&#39; ,reducedDim = &#39;umap&#39;,extend=&quot;n&quot;,stretch=0.07,thresh=0.05)
pto_PRC &lt;- as.SlingshotDataSet(pto_PRC)

plot(dimred, col = palet_new[clustering], asp = 1, pch = 16)
lines(pto_PRC@curves$Lineage3, lwd = 3, col = &#39;magenta&#39;)
lines(pto_TRC@curves$Lineage3, lwd = 3, col = &#39;gold&#39;)
lines(pto_TRC@curves$Lineage1, lwd = 3, col = &#39;black&#39;)
legend(&quot;bottomright&quot;, legend = c(&quot;T3&quot;), col = c(&#39;gold&#39;), lty = 1, lwd = 2.5, cex = 1.1)
legend(&quot;topright&quot;, legend = names(cols), col = cols, pch = 2, pt.cex = 1, cex = 0.8)</code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/CCL19%20EYFP%20mCOV-FIt31-g33%20differentiation%20trajectories-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-CCL19-EYFP-mCOV-FIt31-g33-differentiation-trajectories-1">
Past versions of "CCL19 EYFP mCOV-FIt31-g33 differentiation
trajectories-1.png"
</button>
</p>
<div
id="fig-no-spaces-CCL19-EYFP-mCOV-FIt31-g33-differentiation-trajectories-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/CCL19 EYFP mCOV-FIt31-g33 differentiation trajectories-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="combine-gene-signatures-in-t3-differentiation-trajectory-figure-6h"
class="section level3">
<h3>Combine gene signatures in T3 differentiation trajectory (Figure
6H)</h3>
<pre class="r"><code>counts &lt;- CCL19_EYFP_mCOV@assays$RNA@counts

CCL19_EYFP_mCOV &lt;- FindVariableFeatures(CCL19_EYFP_mCOV, selection.method = &quot;vst&quot;, nfeatures = 1000)

new_counts &lt;-counts[rownames(counts) %in% CCL19_EYFP_mCOV@assays$RNA@var.features,]
sds &lt;- as.SlingshotDataSet(pto_TRC)
pseudotime &lt;- slingPseudotime(pto_TRC, na = FALSE)
cellWeights &lt;- slingCurveWeights(pto_TRC)

BPPARAM &lt;- BiocParallel::bpparam()
BPPARAM$workers &lt;- 4 # use 2 cores
control &lt;- gam.control()
control$maxit &lt;- 1000 </code></pre>
<pre class="r"><code>set.seed(3)
sce &lt;- fitGAM(counts = new_counts, pseudotime = pseudotime, cellWeights = cellWeights,
                 nknots = 6, verbose = TRUE,parallel=TRUE, BPPARAM = BPPARAM, sds = sds,control = control)</code></pre>
</div>
<div id="save-sce-object" class="section level3">
<h3>Save sce object</h3>
<pre class="r"><code>#saveRDS(sce, paste0(basedir,&quot;/data/Mouse/Trajectory_fitGAM_T3_1000.rds&quot;))</code></pre>
</div>
<div id="read-sce-object" class="section level3">
<h3>Read sce object</h3>
<p>We provide the sce object because the fitGAM commnand took time to
run</p>
<pre class="r"><code>sce &lt;- readRDS(paste0(basedir,&quot;/data/Mouse/Trajectory_fitGAM_T3_1000.rds&quot;))</code></pre>
<p>We focus on genes whose expression changes across the T3 trajectory
in the pseudotime. In particular, on those genes that are upregulated in
TLS TRC. We use the startVsEndTest statistical test.</p>
</div>
<div id="gene-expression-changes-in-t3-trajectory"
class="section level3">
<h3>Gene expression changes in T3 trajectory</h3>
<pre class="r"><code>pseudotime_start_end_association &lt;- startVsEndTest(sce) </code></pre>
</div>
<div
id="calculate-log-transformed-counts-and-fitted-values-for-a-particular-gene-across-the-t3-lineage"
class="section level3">
<h3>Calculate log-transformed counts and fitted values for a particular
gene across the T3 lineage</h3>
<pre class="r"><code>data_Ccl21a &lt;-plotSmoothers1(sce, counts, gene = &quot;ENSMUSG00000094686.Ccl21a&quot;, lineagesToPlot = c(3))
data_Srgn &lt;-plotSmoothers1(sce, counts, gene = &quot;ENSMUSG00000020077.Srgn&quot;, lineagesToPlot = c(3)) 
data_Il33 &lt;-plotSmoothers1(sce, counts, gene = &quot;ENSMUSG00000024810.Il33&quot;, lineagesToPlot = c(3)) 
data_Cxcl12 &lt;-plotSmoothers1(sce, counts, gene = &quot;ENSMUSG00000061353.Cxcl12&quot;, lineagesToPlot = c(3)) 
data_Ccl19 &lt;-plotSmoothers1(sce, counts, gene = &quot;ENSMUSG00000071005.Ccl19&quot;, lineagesToPlot = c(3))

data_Ccl21a$gene &lt;- &quot;Ccl21a&quot;
data_Srgn$gene &lt;-&quot;Srgn&quot;
data_Il33$gene &lt;-&quot;Il33&quot;
data_Cxcl12$gene &lt;-&quot;Cxcl12&quot;
data_Ccl19$gene &lt;-&quot;Ccl19&quot;</code></pre>
</div>
<div id="plot-gene-curves" class="section level3">
<h3>Plot gene curves</h3>
<pre class="r"><code>cols_pal &lt;- c(&quot;orange&quot;,&quot;#00B0F6&quot;,&quot;#00BB4E&quot;,&quot;#C09B00&quot;, &quot;#FF6A98&quot;)
names(cols_pal) &lt;-c(&quot;Ccl19&quot;,&quot;Il33&quot;,&quot;Cxcl12&quot;,&quot;Ccl21a&quot;,&quot;Srgn&quot;)

visual12 &lt;- rbind(data_Ccl21a,data_Srgn,data_Il33,data_Cxcl12,data_Ccl19)
end_val &lt;-round(max(visual12$time)) + 1
end_y_axis &lt;-round(max(log1p(visual12$gene_count)))

ggplot(visual12, aes(x=time, y=log1p(gene_count), group=gene, col = gene, fill=gene)) +
  labs(x = &quot;Pseudotime (T3)&quot;, y = &quot;Log(expression + 1)&quot;) +
      geom_line(lwd = 2) +
  scale_y_continuous(expand = expansion(c(0,0)), limits = c(0.0, end_y_axis),breaks = c(0,1,2,3,4,end_y_axis)) +
  scale_x_continuous(limits = c(0.0, end_val + 0.5), expand = expansion(c(0,0)), breaks = c(0,5,end_val)) + 
 theme(aspect.ratio=1.3, axis.text.y = element_text(angle = 0,colour = &quot;black&quot;)) + 
  theme(axis.line = element_line(colour = &quot;black&quot;),
panel.grid.major = element_blank(),
panel.grid.minor = element_blank(),
panel.border = element_blank(),
panel.background = element_blank(),
legend.title = element_blank(),axis.text.x = element_text(angle = 0, vjust = 0.5,colour = &quot;black&quot;)) +
  scale_color_manual(values = cols_pal)+ 
  ggtitle(paste0(&quot;Cd34&quot;, &quot;\U207A &quot;, &quot;AdvFB &gt;&gt; TLS TRC (T3)&quot;))</code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/Step%203%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Step-3-plot-1">
Past versions of "Step 3 plot-1.png"
</button>
</p>
<div id="fig-no-spaces-Step-3-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/Step 3 plot-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="multidimensional-scaling-mds-plot" class="section level3">
<h3>Multidimensional scaling (MDS) plot</h3>
<p>Multidimensional scaling (MDS) visualizes the level of similarity of
variables in a data set. MDS recognizes the structure of the dataset in
2D, as it maintains the pairwise distances between data points.</p>
</div>
<div
id="read-ccl19-eyfp-cell-data-from-naïve-lungs-and-excised-llc-gp33-tumors-on-day-23"
class="section level3">
<h3>Read CCL19-EYFP⁺ cell data from naïve lungs and excised LLC-gp33
tumors on day 23</h3>
<pre class="r"><code>CCL19_EYFP &lt;- readRDS(paste0(basedir,&quot;/data/Mouse/CCL19_EYFP_nonmCOV.rds&quot;))</code></pre>
</div>
<div
id="merge-ccl19-eyfp-cells-from-mcov-fit31-g33-and-non-immunization"
class="section level3">
<h3>Merge CCL19-EYFP⁺ cells from mCOV-FIt31-g33 and non
immunization</h3>
<pre class="r"><code>data_merge &lt;- merge(CCL19_EYFP, y = c(CCL19_EYFP_mCOV),
             add.cell.ids = c(&quot;CCL19_EYFP&quot;,&quot;CCL19_EYFP_mCOV&quot;),
             project = &quot;merge_no_mcov_mcov&quot;)

#Preprocessing
resolution &lt;- c(0.1, 0.25, 0.4, 0.6,0.8, 1.)
data_merge  &lt;- preprocessing(data_merge,resolution)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 511917

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9547
Number of communities: 5
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 511917

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9218
Number of communities: 9
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 511917

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8969
Number of communities: 11
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 511917

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8737
Number of communities: 15
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 511917

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8545
Number of communities: 16
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 511917

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8371
Number of communities: 16
Elapsed time: 1 seconds</code></pre>
</div>
<div
id="integrate-data-to-correct-for-batch-effects-due-to-different-condition-naive-vs-mcov-fit31-g33-via-seurat"
class="section level3">
<h3><strong>Integrate data to correct for batch effects due to different
condition (naive vs mCOV-FIt31-g33) via seurat</strong></h3>
</div>
<div id="step-1" class="section level3">
<h3>Step 1</h3>
<pre class="r"><code>#Split object by Patient as we see batch effects coming from different patients
obj.list &lt;-SplitObject(data_merge, split.by = &#39;condition&#39;)
#For each object in list we see to run normalization and identify highly variable features
for (i in 1:length(obj.list)){
  #Normalization
  obj.list[[i]] &lt;- NormalizeData(obj.list[[i]], normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000) 
  #Find high variable genes
  obj.list[[i]] &lt;- FindVariableFeatures(obj.list[[i]], selection.method = &quot;vst&quot;, nfeatures = 2000)#The purpose of this is to id
}</code></pre>
</div>
<div id="step-2" class="section level3">
<h3>Step 2</h3>
<pre class="r"><code>#select features that are repeatedly variable across datasets for integration
features &lt;- SelectIntegrationFeatures(object.list = obj.list)
#Find anchors to integrate the data across different patients (Canonical correlation analysis)
anchors &lt;- FindIntegrationAnchors(object.list = obj.list, anchor.features = features)
# this command creates an &#39;integrated&#39; data assay
seurat_integrated &lt;- IntegrateData(anchorset = anchors)</code></pre>
</div>
<div id="step-3" class="section level3">
<h3>Step 3</h3>
<pre class="r"><code># We run a single integrated analysis on all cells!
DefaultAssay(seurat_integrated) &lt;- &quot;integrated&quot;

# Run the standard workflow for visualization and clustering
seurat_integrated &lt;- ScaleData(seurat_integrated, verbose = FALSE)
seurat_integrated &lt;- RunPCA(object = seurat_integrated, npcs = 30, verbose = FALSE,seed.use = 8734)
seurat_integrated &lt;- RunTSNE(object = seurat_integrated, reduction = &quot;pca&quot;, dims = 1:20, seed.use = 8734)
seurat_integrated&lt;- RunUMAP(object = seurat_integrated, reduction = &quot;pca&quot;, dims = 1:20, seed.use = 8734)
 seurat_integrated &lt;- FindNeighbors(object = seurat_integrated, reduction = &quot;pca&quot;, dims = 1:20, seed.use = 8734)
#Clustering 
resolution &lt;- c(0.1, 0.25, 0.4, 0.6,0.8, 1.)
for(k in 1:length(resolution)){
  seurat_integrated &lt;- FindClusters(object = seurat_integrated, resolution = resolution[k], random.seed = 8734)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 525485

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9570
Number of communities: 5
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 525485

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9236
Number of communities: 8
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 525485

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8964
Number of communities: 11
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 525485

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8676
Number of communities: 14
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 525485

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8463
Number of communities: 16
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 15462
Number of edges: 525485

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8299
Number of communities: 18
Elapsed time: 1 seconds</code></pre>
</div>
<div id="batch-effects-based-on-condition-are-corrected"
class="section level3 tabset">
<h3 class="tabset">Batch effects based on condition are corrected</h3>
<div id="color-per-condition" class="section level4">
<h4>Color per condition</h4>
<pre class="r"><code>DimPlot(seurat_integrated, reduction = &quot;umap&quot;, group.by = &quot;condition&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;) </code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/umap%20per%20condition-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umap-per-condition-1">
Past versions of "umap per condition-1.png"
</button>
</p>
<div id="fig-no-spaces-umap-per-condition-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/umap per condition-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="existing-cell-type-annotation" class="section level4">
<h4>Existing cell type annotation</h4>
<pre class="r"><code>DimPlot(seurat_integrated, reduction = &quot;umap&quot;, group.by = &quot;annot&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;) </code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/umap%20visualization-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-umap-visualization-1">
Past versions of "umap visualization-1.png"
</button>
</p>
<div id="fig-no-spaces-umap-visualization-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/umap visualization-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="plot-gene-signatures" class="section level3 tabset">
<h3 class="tabset">Plot gene signatures</h3>
<pre class="r"><code>DefaultAssay(seurat_integrated)&lt;- &quot;RNA&quot;

genes &lt;- c(&quot;NPNT&quot;,&quot;HHIP&quot;,&quot;CD34&quot;,&quot;ACTA2&quot;,&quot;SULF1&quot;,&quot;CLU&quot;,&quot;CCL21A&quot;,&quot;CCL19&quot;,&quot;IL33&quot;,&quot;SMOC1&quot;,&quot;CXCL13&quot;,&quot;RGS5&quot;)
genes &lt;-str_to_sentence(genes)

genes &lt;- unlist(lapply(genes, function(x) {
  get_full_gene_name(x,seurat_integrated)
}))</code></pre>
<div id="npnt-hhip-cd34-acta2" class="section level4">
<h4>NPNT HHIP CD34 ACTA2</h4>
<pre class="r"><code>DefaultAssay(seurat_integrated)&lt;- &quot;RNA&quot;

FeaturePlot(seurat_integrated, reduction = &quot;umap&quot;, 
          features = genes[1:4],
          cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
          order = T )+
theme(legend.position=&quot;right&quot;, legend.title=element_text(size=5)) </code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/NPNT%20HHIP%20CD34%20ACTA2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-NPNT-HHIP-CD34-ACTA2-1">
Past versions of "NPNT HHIP CD34 ACTA2-1.png"
</button>
</p>
<div id="fig-no-spaces-NPNT-HHIP-CD34-ACTA2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/NPNT HHIP CD34 ACTA2-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sulf1-clu-ccl21a-ccl19" class="section level4">
<h4>SULF1 CLU CCL21A CCL19</h4>
<pre class="r"><code>DefaultAssay(seurat_integrated)&lt;- &quot;RNA&quot;

FeaturePlot(seurat_integrated, reduction = &quot;umap&quot;, 
          features = genes[5:8],
          cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
          order = T )+
theme(legend.position=&quot;right&quot;, legend.title=element_text(size=5)) </code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/SULF1%20CLU%20CCL21A%20CCL19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-SULF1-CLU-CCL21A-CCL19-1">
Past versions of "SULF1 CLU CCL21A CCL19-1.png"
</button>
</p>
<div id="fig-no-spaces-SULF1-CLU-CCL21A-CCL19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/SULF1 CLU CCL21A CCL19-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="il33-smoc1-cxcl13-rgs5" class="section level4">
<h4>IL33 SMOC1 CXCL13 RGS5</h4>
<pre class="r"><code>DefaultAssay(seurat_integrated)&lt;- &quot;RNA&quot;

FeaturePlot(seurat_integrated, reduction = &quot;umap&quot;, 
          features = genes[9:12],
          cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
          order = T )+
theme(legend.position=&quot;right&quot;, legend.title=element_text(size=5)) </code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/IL33%20SMOC1%20CXCL13%20RGS5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-IL33-SMOC1-CXCL13-RGS5-1">
Past versions of "IL33 SMOC1 CXCL13 RGS5-1.png"
</button>
</p>
<div id="fig-no-spaces-IL33-SMOC1-CXCL13-RGS5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/IL33 SMOC1 CXCL13 RGS5-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div
id="perform-new-cell-type-annotation-based-on-the-previous-gene-signatures"
class="section level3">
<h3>Perform new cell type annotation based on the previous gene
signatures</h3>
<pre class="r"><code>DefaultAssay(seurat_integrated) &lt;- &quot;integrated&quot;

seurat_integrated@meta.data$new_annot &lt;--1
seurat_integrated$new_annot[which(seurat_integrated$integrated_snn_res.0.25 == 0)] &lt;- paste0(&quot;Cd34&quot;, &quot;\u207A &quot;, &quot;AdvFB&quot;)

seurat_integrated$new_annot[which(seurat_integrated$integrated_snn_res.0.25 == 1)] &lt;- paste0(&quot;Hhip&quot;, &quot;\u207A &quot;, &quot;AdvFB&quot;)

seurat_integrated$new_annot[which(seurat_integrated$integrated_snn_res.0.25 == 2)] &lt;- paste0(&quot;Npnt&quot;, &quot;\u207A &quot;, &quot;AlvFB&quot;)

seurat_integrated$new_annot[which(seurat_integrated$integrated_snn_res.0.25 == 3)] &lt;- &quot;SMC/PC&quot;

seurat_integrated$new_annot[which(seurat_integrated$integrated_snn_res.0.25 == 4)] &lt;- paste0(&quot;Sulf1&quot;, &quot;\u207A &quot;, &quot;TRC&quot;)

seurat_integrated$new_annot[which(seurat_integrated$integrated_snn_res.0.25 == 5)] &lt;- paste0(&quot;Rgs5&quot;, &quot;\u207A &quot;, &quot;PRC&quot;)

seurat_integrated$new_annot[which(seurat_integrated$integrated_snn_res.0.25 == 6)] &lt;- paste0(&quot;Npnt&quot;, &quot;\u207A &quot;, &quot;AlvFB&quot;)

seurat_integrated$new_annot[which(seurat_integrated$integrated_snn_res.0.25 == 7)] &lt;- paste0(&quot;TLS &quot;, &quot;TRC&quot;)</code></pre>
</div>
<div id="visualize-new-cell-type-annotation" class="section level3">
<h3>Visualize new cell type annotation</h3>
<pre class="r"><code>DimPlot(seurat_integrated, reduction = &quot;umap&quot;, group.by = &quot;new_annot&quot;, cols = cols)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;) #+ NoLegend()</code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/plot%20new%20cell%20type%20annotation-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-plot-new-cell-type-annotation-1">
Past versions of "plot new cell type annotation-1.png"
</button>
</p>
<div id="fig-no-spaces-plot-new-cell-type-annotation-1"
class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/plot new cell type annotation-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mds-computation" class="section level3">
<h3>MDS computation</h3>
<pre class="r"><code># Before running MDS, we first calculate a distance matrix between all pairs of cells.
# Here we use a simple euclidean distance metric on all genes, using scale.data as input
d &lt;- dist(t(GetAssayData(seurat_integrated, slot = &#39;scale.data&#39;)))
d_mat &lt;-as.matrix(d)
# Run the MDS procedure, k determines the number of dimensions
mds &lt;- cmdscale(d = d, k = 2)
# cmdscale returns the cell embeddings, we first label the columns to ensure downstream consistency
colnames(mds) &lt;- paste0(&quot;MDS_&quot;, 1:2)
# We will now store this as a custom dimensional reduction called &quot;mds&quot;
seurat_integrated[[&#39;mds&#39;]] &lt;- CreateDimReducObject(embeddings = mds, key = &#39;MDS_&#39;, assay = DefaultAssay(seurat_integrated))</code></pre>
<pre class="r"><code>#saveRDS(seurat_integrated, paste0(basedir,&quot;/data/Mouse/integrated_naive_mcov_mds.rds&quot;))</code></pre>
<p>We provide the integrated object with MDS representation as the
classical MDS algorithm takes a long time to run.</p>
</div>
<div id="read-integrated-object-with-mds-representation"
class="section level3">
<h3>Read integrated object with MDS representation</h3>
<pre class="r"><code>seurat_integrated &lt;- readRDS(paste0(basedir,&quot;/data/Mouse/integrated_naive_mcov_mds.rds&quot;))</code></pre>
</div>
<div id="visualize-mds-plot-figure-6e" class="section level3">
<h3>Visualize MDS plot (Figure 6E)</h3>
<pre class="r"><code>mds_tx_condition = seurat_integrated@reductions$mds@cell.embeddings %&gt;% 
as.data.frame() %&gt;% cbind(tx = seurat_integrated@meta.data$condition)

mds_tx_celltype = seurat_integrated@reductions$mds@cell.embeddings %&gt;% 
as.data.frame() %&gt;% cbind(tx = seurat_integrated@meta.data$new_annot)

mds_tx_TOTAL &lt;-merge(mds_tx_condition, mds_tx_celltype, by=c(&quot;MDS_1&quot;, &quot;MDS_2&quot;), all.x=T, all.y=T)
colnames(mds_tx_TOTAL) &lt;-c(&quot;MDS_1&quot;, &quot;MDS_2&quot;, &quot;Condition&quot;,&quot;Celltype&quot;)

mds_tx_TOTAL &lt;- mds_tx_TOTAL %&gt;% 
  group_by(Celltype,Condition) %&gt;% 
  mutate(count_mds1 = mean(MDS_1)) %&gt;%
  mutate(count_mds2 = mean(MDS_2))

test &lt;-mds_tx_TOTAL %&gt;%
  group_by(Celltype) %&gt;%
  filter(!(Celltype == &#39;TLS TRC&#39; &amp; Condition == &#39;Not immunized&#39;))

ggplot(test, aes(x=count_mds1, y=count_mds2, color=Celltype, shape = Condition)) + geom_point(stroke = 1.5) + ylab(&quot;MDS2&quot;) + xlab(&quot;MDS1&quot;) +  
scale_color_manual(values=cols) + scale_shape_manual(values = c(2, 3)) + 
  theme(aspect.ratio = 1.3, axis.line = element_line(colour = &quot;black&quot;),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank(),
    panel.background = element_blank(),
    axis.text.y = element_text(angle = 0, vjust = 0.5,colour = &quot;black&quot;,size = 10),
    axis.text.x = element_text(angle = 0, vjust = 0.5,colour = &quot;black&quot;,size = 10),
    
) </code></pre>
<p><img src="figure/Mouse_mCOV.Rmd/MDS%20visualization-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-MDS-visualization-1">
Past versions of "MDS visualization-1.png"
</button>
</p>
<div id="fig-no-spaces-MDS-visualization-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Pchryssa/CCL19_FRCs_lung_cancer/blob/80d46cf78223b1660290f3fefa56ae7d3f101135/docs/figure/Mouse_mCOV.Rmd/MDS visualization-1.png" target="_blank">80d46cf</a>
</td>
<td>
Pchryssa
</td>
<td>
2024-08-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="session-info" class="section level3">
<h3>Session info</h3>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.6.9

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] mgcv_1.9-0                  nlme_3.1-162               
 [3] tradeSeq_1.14.0             slingshot_2.8.0            
 [5] TrajectoryUtils_1.8.0       SingleCellExperiment_1.22.0
 [7] SummarizedExperiment_1.30.2 Biobase_2.60.0             
 [9] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        
[11] IRanges_2.34.1              S4Vectors_0.38.1           
[13] BiocGenerics_0.46.0         MatrixGenerics_1.12.3      
[15] matrixStats_1.0.0           princurve_2.1.6            
[17] ggsci_3.0.0                 gsubfn_0.7                 
[19] proto_1.0.0                 gridExtra_2.3              
[21] dittoSeq_1.12.1             ggplot2_3.4.2              
[23] Matrix_1.6-0                SeuratObject_4.1.3         
[25] Seurat_4.3.0.1              patchwork_1.1.2            
[27] stringr_1.5.0               dplyr_1.1.2                
[29] purrr_1.0.1                 here_1.0.1                 
[31] magrittr_2.0.3              circlize_0.4.15            
[33] tidyr_1.3.0                 tibble_3.2.1               
[35] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21          splines_4.3.1            
  [3] later_1.3.1               bitops_1.0-7             
  [5] polyclip_1.10-4           lifecycle_1.0.3          
  [7] tcltk_4.3.1               edgeR_3.42.4             
  [9] rprojroot_2.0.3           globals_0.16.2           
 [11] processx_3.8.2            lattice_0.21-8           
 [13] MASS_7.3-60               limma_3.56.2             
 [15] plotly_4.10.2             sass_0.4.7               
 [17] rmarkdown_2.23            jquerylib_0.1.4          
 [19] yaml_2.3.7                httpuv_1.6.11            
 [21] sctransform_0.3.5         sp_2.0-0                 
 [23] spatstat.sparse_3.0-2     reticulate_1.36.1        
 [25] cowplot_1.1.1             pbapply_1.7-2            
 [27] RColorBrewer_1.1-3        abind_1.4-5              
 [29] zlibbioc_1.46.0           Rtsne_0.16               
 [31] RCurl_1.98-1.12           git2r_0.33.0             
 [33] GenomeInfoDbData_1.2.10   ggrepel_0.9.3            
 [35] irlba_2.3.5.1             listenv_0.9.0            
 [37] spatstat.utils_3.1-0      pheatmap_1.0.12          
 [39] goftest_1.2-3             spatstat.random_3.1-5    
 [41] fitdistrplus_1.1-11       parallelly_1.36.0        
 [43] DelayedMatrixStats_1.22.0 leiden_0.4.3             
 [45] codetools_0.2-19          DelayedArray_0.28.0      
 [47] tidyselect_1.2.0          shape_1.4.6              
 [49] farver_2.1.1              viridis_0.6.4            
 [51] spatstat.explore_3.2-1    jsonlite_1.8.7           
 [53] ellipsis_0.3.2            progressr_0.13.0         
 [55] ggridges_0.5.4            survival_3.5-5           
 [57] systemfonts_1.0.4         tools_4.3.1              
 [59] ragg_1.2.5                ica_1.0-3                
 [61] Rcpp_1.0.11               glue_1.6.2               
 [63] SparseArray_1.2.4         xfun_0.39                
 [65] withr_2.5.0               fastmap_1.1.1            
 [67] fansi_1.0.4               callr_3.7.3              
 [69] digest_0.6.33             R6_2.5.1                 
 [71] mime_0.12                 textshaping_0.3.6        
 [73] colorspace_2.1-0          scattermore_1.2          
 [75] tensor_1.5                spatstat.data_3.0-1      
 [77] utf8_1.2.3                generics_0.1.3           
 [79] data.table_1.14.8         httr_1.4.6               
 [81] htmlwidgets_1.6.2         S4Arrays_1.2.1           
 [83] whisker_0.4.1             uwot_0.1.16              
 [85] pkgconfig_2.0.3           gtable_0.3.3             
 [87] lmtest_0.9-40             XVector_0.40.0           
 [89] htmltools_0.5.5           scales_1.2.1             
 [91] png_0.1-8                 knitr_1.43               
 [93] rstudioapi_0.15.0         reshape2_1.4.4           
 [95] cachem_1.0.8              zoo_1.8-12               
 [97] GlobalOptions_0.1.2       KernSmooth_2.23-22       
 [99] parallel_4.3.1            miniUI_0.1.1.1           
[101] pillar_1.9.0              grid_4.3.1               
[103] vctrs_0.6.3               RANN_2.6.1               
[105] promises_1.2.0.1          xtable_1.8-4             
[107] cluster_2.1.4             evaluate_0.21            
[109] locfit_1.5-9.8            cli_3.6.1                
[111] compiler_4.3.1            rlang_1.1.1              
[113] crayon_1.5.2              future.apply_1.11.0      
[115] labeling_0.4.2            ps_1.7.5                 
[117] getPass_0.2-4             plyr_1.8.8               
[119] fs_1.6.3                  stringi_1.7.12           
[121] viridisLite_0.4.2         deldir_1.0-9             
[123] BiocParallel_1.34.2       munsell_0.5.0            
[125] lazyeval_0.2.2            spatstat.geom_3.2-4      
[127] sparseMatrixStats_1.12.2  future_1.33.0            
[129] shiny_1.7.4.1             highr_0.10               
[131] ROCR_1.0-11               igraph_1.5.0.1           
[133] bslib_0.5.0              </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Mon Sep 23 13:03:57 2024&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.6.9

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] mgcv_1.9-0                  nlme_3.1-162               
 [3] tradeSeq_1.14.0             slingshot_2.8.0            
 [5] TrajectoryUtils_1.8.0       SingleCellExperiment_1.22.0
 [7] SummarizedExperiment_1.30.2 Biobase_2.60.0             
 [9] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        
[11] IRanges_2.34.1              S4Vectors_0.38.1           
[13] BiocGenerics_0.46.0         MatrixGenerics_1.12.3      
[15] matrixStats_1.0.0           princurve_2.1.6            
[17] ggsci_3.0.0                 gsubfn_0.7                 
[19] proto_1.0.0                 gridExtra_2.3              
[21] dittoSeq_1.12.1             ggplot2_3.4.2              
[23] Matrix_1.6-0                SeuratObject_4.1.3         
[25] Seurat_4.3.0.1              patchwork_1.1.2            
[27] stringr_1.5.0               dplyr_1.1.2                
[29] purrr_1.0.1                 here_1.0.1                 
[31] magrittr_2.0.3              circlize_0.4.15            
[33] tidyr_1.3.0                 tibble_3.2.1               
[35] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21          splines_4.3.1            
  [3] later_1.3.1               bitops_1.0-7             
  [5] polyclip_1.10-4           lifecycle_1.0.3          
  [7] tcltk_4.3.1               edgeR_3.42.4             
  [9] rprojroot_2.0.3           globals_0.16.2           
 [11] processx_3.8.2            lattice_0.21-8           
 [13] MASS_7.3-60               limma_3.56.2             
 [15] plotly_4.10.2             sass_0.4.7               
 [17] rmarkdown_2.23            jquerylib_0.1.4          
 [19] yaml_2.3.7                httpuv_1.6.11            
 [21] sctransform_0.3.5         sp_2.0-0                 
 [23] spatstat.sparse_3.0-2     reticulate_1.36.1        
 [25] cowplot_1.1.1             pbapply_1.7-2            
 [27] RColorBrewer_1.1-3        abind_1.4-5              
 [29] zlibbioc_1.46.0           Rtsne_0.16               
 [31] RCurl_1.98-1.12           git2r_0.33.0             
 [33] GenomeInfoDbData_1.2.10   ggrepel_0.9.3            
 [35] irlba_2.3.5.1             listenv_0.9.0            
 [37] spatstat.utils_3.1-0      pheatmap_1.0.12          
 [39] goftest_1.2-3             spatstat.random_3.1-5    
 [41] fitdistrplus_1.1-11       parallelly_1.36.0        
 [43] DelayedMatrixStats_1.22.0 leiden_0.4.3             
 [45] codetools_0.2-19          DelayedArray_0.28.0      
 [47] tidyselect_1.2.0          shape_1.4.6              
 [49] farver_2.1.1              viridis_0.6.4            
 [51] spatstat.explore_3.2-1    jsonlite_1.8.7           
 [53] ellipsis_0.3.2            progressr_0.13.0         
 [55] ggridges_0.5.4            survival_3.5-5           
 [57] systemfonts_1.0.4         tools_4.3.1              
 [59] ragg_1.2.5                ica_1.0-3                
 [61] Rcpp_1.0.11               glue_1.6.2               
 [63] SparseArray_1.2.4         xfun_0.39                
 [65] withr_2.5.0               fastmap_1.1.1            
 [67] fansi_1.0.4               callr_3.7.3              
 [69] digest_0.6.33             R6_2.5.1                 
 [71] mime_0.12                 textshaping_0.3.6        
 [73] colorspace_2.1-0          scattermore_1.2          
 [75] tensor_1.5                spatstat.data_3.0-1      
 [77] utf8_1.2.3                generics_0.1.3           
 [79] data.table_1.14.8         httr_1.4.6               
 [81] htmlwidgets_1.6.2         S4Arrays_1.2.1           
 [83] whisker_0.4.1             uwot_0.1.16              
 [85] pkgconfig_2.0.3           gtable_0.3.3             
 [87] lmtest_0.9-40             XVector_0.40.0           
 [89] htmltools_0.5.5           scales_1.2.1             
 [91] png_0.1-8                 knitr_1.43               
 [93] rstudioapi_0.15.0         reshape2_1.4.4           
 [95] cachem_1.0.8              zoo_1.8-12               
 [97] GlobalOptions_0.1.2       KernSmooth_2.23-22       
 [99] parallel_4.3.1            miniUI_0.1.1.1           
[101] pillar_1.9.0              grid_4.3.1               
[103] vctrs_0.6.3               RANN_2.6.1               
[105] promises_1.2.0.1          xtable_1.8-4             
[107] cluster_2.1.4             evaluate_0.21            
[109] locfit_1.5-9.8            cli_3.6.1                
[111] compiler_4.3.1            rlang_1.1.1              
[113] crayon_1.5.2              future.apply_1.11.0      
[115] labeling_0.4.2            ps_1.7.5                 
[117] getPass_0.2-4             plyr_1.8.8               
[119] fs_1.6.3                  stringi_1.7.12           
[121] viridisLite_0.4.2         deldir_1.0-9             
[123] BiocParallel_1.34.2       munsell_0.5.0            
[125] lazyeval_0.2.2            spatstat.geom_3.2-4      
[127] sparseMatrixStats_1.12.2  future_1.33.0            
[129] shiny_1.7.4.1             highr_0.10               
[131] ROCR_1.0-11               igraph_1.5.0.1           
[133] bslib_0.5.0              </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
